{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[WDM] - \n",
      "\n",
      "[WDM] - ====== WebDriver manager ======\n",
      "[WDM] - Current google-chrome version is 100.0.4896\n",
      "[WDM] - Get LATEST driver version for 100.0.4896\n",
      "[WDM] - Driver [/Users/gaignard-a/.wdm/drivers/chromedriver/mac64/100.0.4896.60/chromedriver] found in cache\n"
     ]
    }
   ],
   "source": [
    "import sys\n",
    "parentdir = \"..\"\n",
    "sys.path.insert(0, parentdir)\n",
    "\n",
    "import requests\n",
    "from os import path\n",
    "from tqdm.notebook import tqdm\n",
    "import pandas as pd\n",
    "import time\n",
    "\n",
    "from metrics.WebResource import WebResource"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 1. Retrieve a list of target URLs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "target_urls = [\"http://bio.tools/bwa\", \"http://bio.tools/jaspar\", \"https://doi.pangaea.de/10.1594/PANGAEA.942997\", \"https://zenodo.org/record/6411574#.Ykw0bi8itqs\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "#for url in tqdm(target_urls):\n",
    "#    print(url)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 2. Feed the RDF graph"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "from rdflib import ConjunctiveGraph, Namespace, URIRef\n",
    "from rdflib.namespace import RDF, RDFS\n",
    "\n",
    "schema = Namespace(\"http://schema.org/\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "KG = ConjunctiveGraph()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "cc01abdb6973496bbec4ea24663f1bb7",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/4 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded 124 triples\n",
      "Loaded 219 triples\n",
      "http://schema.org/\n",
      "Loaded 380 triples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING:root:Can't find \"application/ld+json\" content\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "https://schema.org/\n",
      "Loaded 440 triples\n",
      "Loaded 440 triples\n"
     ]
    }
   ],
   "source": [
    "for u in tqdm(target_urls):\n",
    "    wr = WebResource(url=u)\n",
    "    KG += wr.get_rdf()\n",
    "    print(f\"Loaded {len(KG)} triples\")\n",
    "\n",
    "print(f\"Loaded {len(KG)} triples\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Graph identifier=N50cfbbd444fe4d2999d0ec02e19be206 (<class 'rdflib.graph.ConjunctiveGraph'>)>"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "KG.serialize(\"scrapped_bioschemas.ttl\", format=\"turtle\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@prefix ns1: <http://schema.org/> .\r\n",
      "@prefix ns2: <http://bioschemas.org/> .\r\n",
      "@prefix ns3: <http://ogp.me/ns#> .\r\n",
      "@prefix ns4: <https://bio.tools/ontology/> .\r\n",
      "@prefix ns5: <http://purl.org/dc/terms/> .\r\n",
      "@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .\r\n",
      "\r\n",
      "<file:///Users/gaignard-a/Documents/Dev/fair-checker/notebooks/> ns3:description \"This dataset contains impact metrics and indicators for a set of publications that are related to the COVID-19 infectious disease and the coronavirus that causes it. It is based on: Τhe CORD-19 dataset released by the team of Semantic Scholar1 and Τhe curated data provided by the LitCovid hub2. These data have been cleaned and integrated with data from COVID-19-TweetIDs and from other sources (e.g., PMC). The result was dataset of 503,979 unique articles along with relevant metadata (e.g., the underlying citation network). We utilized this dataset to produce, for each article, the values of the following impact measures: Influence: Citation-based measure reflecting the total impact of an article. This is based on the PageRank3 network analysis method. In the context of citation networks, it estimates the importance of each article based on its centrality in the whole network. This measure was calculated using the PaperRanking (https://github.com/diwis/PaperRanking) library4. Influence_alt: Citation-based measure reflecting the total impact of an article. This is the Citation Count of each article, calculated based on the citation network between the articles contained in the BIP4COVID19 dataset. Popularity: Citation-based measure reflecting the current impact of an article. This is based on the AttRank5 citation network analysis method. Methods like PageRank are biased against recently published articles (new articles need time to receive their first citations). AttRank alleviates this problem incorporating an attention-based mechanism, akin to a time-restricted version of preferential attachment, to explicitly capture a researcher's preference to read papers which received a lot of attention recently. This is why it is more suitable to capture the current \\\"hype\\\" of an article. Popularity alternative: An alternative citation-based measure reflecting the current impact of an article (this was the basic popularity measured provided by BIP4COVID19 until version 26). This is based on the RAM6 citation network analysis method. Methods like PageRank are biased against recently published articles (new articles need time to receive their first citations). RAM alleviates this problem using an approach known as \\\"time-awareness\\\". This is why it is more suitable to capture the current \\\"hype\\\" of an article. This measure was calculated using the PaperRanking (https://github.com/diwis/PaperRanking) library4. Social Media Attention: The number of tweets related to this article. Relevant data were collected from the COVID-19-TweetIDs dataset. In this version, tweets between 5/3/22-12/3/22 have been considered from the previous dataset.  We provide five CSV files, all containing the same information, however each having its entries ordered by a different impact measure. All CSV files are tab separated and have the same columns (PubMed_id, PMC_id, DOI, influence_score, popularity_alt_score, popularity score, influence_alt score, tweets count). The work is based on the following publications: COVID-19 Open Research Dataset (CORD-19). 2020. Version 2022-03-28 Retrieved from https://pages.semanticscholar.org/coronavirus-research. Accessed 2022-03-28. doi:10.5281/zenodo.3715506 Chen Q, Allot A, & Lu Z. (2020) Keep up with the latest coronavirus research, Nature 579:193 (version 2022-03-28) R. Motwani L. Page, S. Brin and T. Winograd. 1999. The PageRank Citation Ranking: Bringing Order to the Web. Technical Report. Stanford InfoLab. I. Kanellos, T. Vergoulis, D. Sacharidis, T. Dalamagas, Y. Vassiliou: Impact-Based Ranking of Scientific Publications: A Survey and Experimental Evaluation. TKDE 2019 I. Kanellos, T. Vergoulis, D. Sacharidis, T. Dalamagas, Y. Vassiliou: Ranking Papers by their Short-Term Scientific Impact. CoRR abs/2006.00951 (2020) Rumi Ghosh, Tsung-Ting Kuo, Chun-Nan Hsu, Shou-De Lin, and Kristina Lerman. 2011. Time-Aware Ranking in Dynamic Citation Networks. In Data Mining Workshops (ICDMW). 373–380 A Web user interface that uses these data to facilitate the COVID-19 literature exploration, can be found here. More details in our peer-reviewed publication here (also here there is an outdated preprint version). Funding: We acknowledge support of this work by the project \\\"Moving from Big Data Management to Data Science\\\" (MIS 5002437/3) which is implemented under the Action \\\"Reinforcement of the Research and Innovation Infrastructure\\\", funded by the Operational Programme \\\"Competitiveness, Entrepreneurship and Innovation\\\" (NSRF 2014-2020) and co-financed by Greece and the European Union (European Regional Development Fund). Terms of use: These data are provided \\\"as is\\\", without any warranties of any kind. The data are provided under the Creative Commons Attribution 4.0 International license.\",\r\n",
      "        \"{{ngMeta['og:description']}}\" ;\r\n",
      "    ns3:image \"https://www.pangaea.de/assets/social-icons/pangaea-share.png\",\r\n",
      "        \"{{ngMeta['og:image']}}\" ;\r\n",
      "    ns3:site_name \"Zenodo\" ;\r\n",
      "    ns3:title \"BIP4COVID19: Impact metrics and indicators for coronavirus related publications\",\r\n",
      "        \"{{ngMeta['og:title']}}\" ;\r\n",
      "    ns3:url \"https://zenodo.org/record/6411574\" .\r\n",
      "\r\n",
      "<https://bio.tools/bwa> a ns1:SoftwareApplication ;\r\n",
      "    ns2:input <https://bio.tools/bwa/op_1/in_1>,\r\n",
      "        <https://bio.tools/bwa/op_2/in_1>,\r\n",
      "        <https://bio.tools/bwa/op_2/in_2>,\r\n",
      "        <https://bio.tools/bwa/op_3/in_1>,\r\n",
      "        <https://bio.tools/bwa/op_3/in_2>,\r\n",
      "        <https://bio.tools/bwa/op_4/in_1>,\r\n",
      "        <https://bio.tools/bwa/op_4/in_2>,\r\n",
      "        <https://bio.tools/bwa/op_4/in_3>,\r\n",
      "        <https://bio.tools/bwa/op_5/in_1>,\r\n",
      "        <https://bio.tools/bwa/op_5/in_2>,\r\n",
      "        <https://bio.tools/bwa/op_5/in_3>,\r\n",
      "        <https://bio.tools/bwa/op_6/in_1>,\r\n",
      "        <https://bio.tools/bwa/op_6/in_2> ;\r\n",
      "    ns2:output <https://bio.tools/bwa/op_1/in_1>,\r\n",
      "        <https://bio.tools/bwa/op_2/in_1>,\r\n",
      "        <https://bio.tools/bwa/op_3/in_1>,\r\n",
      "        <https://bio.tools/bwa/op_4/in_1>,\r\n",
      "        <https://bio.tools/bwa/op_5/in_1>,\r\n",
      "        <https://bio.tools/bwa/op_6/in_1> ;\r\n",
      "    ns5:conformsTo \"https://bioschemas.org/profiles/ComputationalTool/1.0-RELEASE\" ;\r\n",
      "    ns1:applicationSubCategory <http://edamontology.org/topic_0102> ;\r\n",
      "    ns1:citation <https://doi.org/10.1016/j.ygeno.2017.03.001>,\r\n",
      "        <https://doi.org/10.1093/bioinformatics/btp324>,\r\n",
      "        <https://doi.org/10.1093/bioinformatics/btp698>,\r\n",
      "        <https://doi.org/10.1093/bioinformatics/btu146>,\r\n",
      "        <https://doi.org/10.1186/1471-2105-14-184>,\r\n",
      "        <https://doi.org/10.1186/1471-2164-15-264>,\r\n",
      "        \"pmcid:PMC2705234\",\r\n",
      "        \"pmcid:PMC3694458\",\r\n",
      "        \"pmcid:PMC4051166\",\r\n",
      "        \"pubmed:19451168\",\r\n",
      "        \"pubmed:20080505\",\r\n",
      "        \"pubmed:23758764\",\r\n",
      "        \"pubmed:24626854\",\r\n",
      "        \"pubmed:24708189\",\r\n",
      "        \"pubmed:28286147\" ;\r\n",
      "    ns1:contributor <http://orcid.org/0000-0003-4874-2874> ;\r\n",
      "    ns1:featureList <http://edamontology.org/operation_0292>,\r\n",
      "        <http://edamontology.org/operation_3198>,\r\n",
      "        <http://edamontology.org/operation_3211>,\r\n",
      "        <http://edamontology.org/operation_3429> ;\r\n",
      "    ns1:softwareHelp <http://bio-bwa.sourceforge.net/bwa.shtml> ;\r\n",
      "    ns4:primaryContact \"bwa team\" .\r\n",
      "\r\n",
      "<https://bio.tools/jaspar> a ns1:SoftwareApplication ;\r\n",
      "    ns2:input <https://bio.tools/jaspar/op_1/in_1>,\r\n",
      "        <https://bio.tools/jaspar/op_1/in_2>,\r\n",
      "        <https://bio.tools/jaspar/op_1/in_3>,\r\n",
      "        <https://bio.tools/jaspar/op_1/in_4>,\r\n",
      "        <https://bio.tools/jaspar/op_1/in_5>,\r\n",
      "        <https://bio.tools/jaspar/op_1/in_6>,\r\n",
      "        <https://bio.tools/jaspar/op_1/in_7>,\r\n",
      "        <https://bio.tools/jaspar/op_2/in_1>,\r\n",
      "        <https://bio.tools/jaspar/op_2/in_2> ;\r\n",
      "    ns2:output <https://bio.tools/jaspar/op_1/in_1>,\r\n",
      "        <https://bio.tools/jaspar/op_2/in_1> ;\r\n",
      "    ns5:conformsTo \"https://bioschemas.org/profiles/ComputationalTool/1.0-RELEASE\" ;\r\n",
      "    ns1:applicationSubCategory <http://edamontology.org/topic_0204>,\r\n",
      "        <http://edamontology.org/topic_0621>,\r\n",
      "        <http://edamontology.org/topic_0622>,\r\n",
      "        <http://edamontology.org/topic_0749>,\r\n",
      "        <http://edamontology.org/topic_0780>,\r\n",
      "        <http://edamontology.org/topic_2815> ;\r\n",
      "    ns1:citation <https://doi.org/10.1093/bioinformatics/btx804>,\r\n",
      "        <https://doi.org/10.1093/nar/gkh012>,\r\n",
      "        <https://doi.org/10.1093/nar/gkj115>,\r\n",
      "        <https://doi.org/10.1093/nar/gkm955>,\r\n",
      "        <https://doi.org/10.1093/nar/gkp950>,\r\n",
      "        <https://doi.org/10.1093/nar/gkt997>,\r\n",
      "        <https://doi.org/10.1093/nar/gkv1176>,\r\n",
      "        <https://doi.org/10.1093/nar/gkx1126>,\r\n",
      "        \"pubmed:14681366\",\r\n",
      "        \"pubmed:16381983\",\r\n",
      "        \"pubmed:18006571\",\r\n",
      "        \"pubmed:19906716\",\r\n",
      "        \"pubmed:24194598\",\r\n",
      "        \"pubmed:26531826\",\r\n",
      "        \"pubmed:29253085\" ;\r\n",
      "    ns1:featureList <http://edamontology.org/operation_0445>,\r\n",
      "        <http://edamontology.org/operation_2421> ;\r\n",
      "    ns1:isAccessibleForFree true ;\r\n",
      "    ns1:softwareHelp <http://jaspar.genereg.net/api/v1/docs/>,\r\n",
      "        <http://jaspar.genereg.net/docs/>,\r\n",
      "        <http://jaspar.genereg.net/faq/> ;\r\n",
      "    ns4:primaryContact <http://orcid.org/0000-0001-5127-5459>,\r\n",
      "        \"Albin Sandelin\",\r\n",
      "        \"Boris Lenhard\",\r\n",
      "        \"Wyeth Wasserman\" .\r\n",
      "\r\n",
      "<https://doi.org/10.5281/zenodo.6411574> a ns1:Dataset ;\r\n",
      "    ns1:citation <https://github.com/diwis/PaperRanking>,\r\n",
      "        <https://pages.semanticscholar.org/coronavirus-research> ;\r\n",
      "    ns1:creator <https://orcid.org/0000-0002-3154-6212>,\r\n",
      "        <https://orcid.org/0000-0002-5002-7901>,\r\n",
      "        <https://orcid.org/0000-0003-0555-4128>,\r\n",
      "        <https://orcid.org/0000-0003-1714-5225>,\r\n",
      "        <https://orcid.org/0000-0003-2146-3795> ;\r\n",
      "    ns1:datePublished \"2022-04-04\"^^ns1:Date ;\r\n",
      "    ns1:description \"\"\"<p>This dataset contains impact metrics and indicators for a set of publications that are related to the <a href=\"https://en.wikipedia.org/wiki/Coronavirus_disease_2019\">COVID-19 infectious disease</a> and the coronavirus that causes it. It is based on:</p>\r\n",
      "\r\n",
      "<ol>\r\n",
      "\t<li>&Tau;he <a href=\"https://pages.semanticscholar.org/coronavirus-research\">CORD-19 dataset</a> released by the team of <a href=\"https://www.semanticscholar.org/\">Semantic Scholar</a><sup>1</sup> and</li>\r\n",
      "\t<li>&Tau;he curated data provided by the <a href=\"https://www.ncbi.nlm.nih.gov/research/coronavirus/\">LitCovid hub</a><sup>2</sup>.</li>\r\n",
      "</ol>\r\n",
      "\r\n",
      "<p>These data have been cleaned and integrated with data from <a href=\"https://github.com/echen102/COVID-19-TweetIDs\">COVID-19-TweetIDs</a> and from other sources (e.g., PMC). The result was dataset of&nbsp;503,979 unique articles along with relevant metadata (e.g., the underlying citation network). We utilized this dataset to produce, for each article, the values of the following impact measures:</p>\r",
      "\r\n",
      "\r\n",
      "<ul>\r\n",
      "\t<li><em><strong>Influence:</strong></em> Citation-based measure reflecting the total impact of an article. This is based on the PageRank<sup>3</sup> network analysis method. In the context of citation networks, it estimates the importance of each article based on its centrality in the whole network. This measure was calculated using the PaperRanking (<a href=\"https://github.com/diwis/PaperRanking\">https://github.com/diwis/PaperRanking</a>) library<sup>4</sup>.</li>\r\n",
      "\t<li><strong><em>Influence_alt:</em></strong> Citation-based measure reflecting the total impact of an article. This is the Citation Count of each article, calculated based on the citation network between the articles contained in the BIP4COVID19 dataset.</li>\r\n",
      "\t<li><em><strong>Popularity:</strong></em> Citation-based measure reflecting the current impact of an article. This is based on the AttRank<sup>5</sup> citation network analysis method. Methods like PageRank are biased against recently published articles (new articles need time to receive their first citations). AttRank alleviates this problem incorporating an attention-based mechanism, akin to a time-restricted version of preferential attachment, to explicitly capture a researcher&#39;s preference to read papers which received a lot of attention recently. This is why it is more suitable to capture the current &quot;hype&quot; of an article.</li>\r\n",
      "\t<li><em><strong>Popularity alternative:</strong></em> An alternative citation-based measure reflecting the current impact of an article (this was the basic popularity measured provided by BIP4COVID19 until version 26). This is based on the RAM<sup>6</sup> citation network analysis method. Methods like PageRank are biased against recently published articles (new articles need time to receive their first citations). RAM alleviates this problem using an approach known as &quot;time-awareness&quot;. This is why it is more suitable to capture the current &quot;hype&quot; of an article. This measure was calculated using the PaperRanking (<a href=\"https://github.com/diwis/PaperRanking\">https://github.com/diwis/PaperRanking</a>) library<sup>4</sup>.</li>\r\n",
      "\t<li><em><strong>Social Media Attention: </strong></em>The number of tweets related to this article. Relevant data were collected from the <a href=\"https://github.com/echen102/COVID-19-TweetIDs\">COVID-19-TweetIDs</a> dataset. In this version, tweets between 5/3/22-12/3/22 have been considered from the previous dataset.&nbsp;</li>\r\n",
      "</ul>\r\n",
      "\r\n",
      "<p>We provide five CSV files, all containing the same information, however each having its entries ordered by a different impact measure. All CSV files are tab separated and have the same columns (PubMed_id, PMC_id, DOI, influence_score, popularity_alt_score, popularity score, influence_alt score, tweets count).</p>\r\n",
      "\r\n",
      "<p>The work is based on the following publications:</p>\r\n",
      "\r\n",
      "<blockquote>\r\n",
      "<ol>\r\n",
      "\t<li>COVID-19 Open Research Dataset (CORD-19). 2020. Version 2022-03-28 Retrieved from https://pages.semanticscholar.org/coronavirus-research. Accessed 2022-03-28. doi:10.5281/zenodo.3715506</li>\r\n",
      "\t<li>Chen Q, Allot A, &amp; Lu Z. (2020) Keep up with the latest coronavirus research, Nature 579:193 (version 2022-03-28)</li>\r\n",
      "\t<li>R. Motwani L. Page, S. Brin and T. Winograd. 1999. The PageRank Citation Ranking: Bringing Order to the Web. Technical Report. Stanford InfoLab.</li>\r\n",
      "\t<li>I. Kanellos, T. Vergoulis, D. Sacharidis, T. Dalamagas, Y. Vassiliou: Impact-Based Ranking of Scientific Publications: A Survey and Experimental Evaluation. TKDE 2019</li>\r\n",
      "\t<li>I. Kanellos, T. Vergoulis, D. Sacharidis, T. Dalamagas, Y. Vassiliou: Ranking Papers by their Short-Term Scientific Impact. CoRR abs/2006.00951 (2020)</li>\r\n",
      "\t<li>Rumi Ghosh, Tsung-Ting Kuo, Chun-Nan Hsu, Shou-De Lin, and Kristina Lerman. 2011. Time-Aware Ranking in Dynamic Citation Networks. In Data Mining Workshops (ICDMW). 373&ndash;380</li>\r\n",
      "</ol>\r\n",
      "</blockquote>\r\n",
      "\r\n",
      "<p>A Web user interface that uses these data to facilitate the COVID-19 literature exploration, can be found <a href=\"https://bip.covid19.athenarc.gr\">here</a>. More details in our peer-reviewed publication <a href=\"https://doi.org/10.1162/qss_a_00169\">here</a> (also <a href=\"https://www.biorxiv.org/content/10.1101/2020.04.11.037093v2\">here</a> there is an outdated preprint version).<br>\r\n",
      "<br>\r\n",
      "<em>Funding: We acknowledge support of this work by the project &quot;Moving from Big Data Management to Data Science&quot; (MIS 5002437/3) which is implemented under the Action &quot;Reinforcement of the Research and Innovation Infrastructure&quot;, funded by the Operational Programme &quot;Competitiveness, Entrepreneurship and Innovation&quot; (NSRF 2014-2020) and co-financed by Greece and the European Union (European Regional Development Fund).</em></p>\r\n",
      "\r\n",
      "<p><em><strong>Terms of use:</strong></em> These data are provided &quot;as is&quot;, without any warranties of any kind. The data are provided under the Creative Commons Attribution 4.0 International license.</p>\"\"\" ;\r\n",
      "    ns1:distribution [ a ns1:DataDownload ;\r\n",
      "            ns1:contentUrl <https://zenodo.org/api/files/da8ab994-0fa3-4d97-b198-779d085977c2/by_popularity_alt.csv> ;\r\n",
      "            ns1:encodingFormat \"csv\" ],\r\n",
      "        [ a ns1:DataDownload ;\r\n",
      "            ns1:contentUrl <https://zenodo.org/api/files/da8ab994-0fa3-4d97-b198-779d085977c2/by_influence.csv> ;\r\n",
      "            ns1:encodingFormat \"csv\" ],\r\n",
      "        [ a ns1:DataDownload ;\r\n",
      "            ns1:contentUrl <https://zenodo.org/api/files/da8ab994-0fa3-4d97-b198-779d085977c2/by_zenodo.csv> ;\r\n",
      "            ns1:encodingFormat \"csv\" ],\r\n",
      "        [ a ns1:DataDownload ;\r\n",
      "            ns1:contentUrl <https://zenodo.org/api/files/da8ab994-0fa3-4d97-b198-779d085977c2/by_popularity.csv> ;\r\n",
      "            ns1:encodingFormat \"csv\" ],\r\n",
      "        [ a ns1:DataDownload ;\r\n",
      "            ns1:contentUrl <https://zenodo.org/api/files/da8ab994-0fa3-4d97-b198-779d085977c2/by_influence_alt.csv> ;\r\n",
      "            ns1:encodingFormat \"csv\" ] ;\r\n",
      "    ns1:identifier \"https://doi.org/10.5281/zenodo.6411574\" ;\r\n",
      "    ns1:keywords \"COVID-19\",\r\n",
      "        \"bibliometrics\",\r\n",
      "        \"coronavirus\",\r\n",
      "        \"scientometrics\" ;\r\n",
      "    ns1:license <https://creativecommons.org/licenses/by/4.0/legalcode> ;\r\n",
      "    ns1:name \"BIP4COVID19: Impact metrics and indicators for coronavirus related publications\" ;\r\n",
      "    ns1:url <https://zenodo.org/record/6411574> ;\r\n",
      "    ns1:version \"87\" .\r\n",
      "\r\n",
      "<http://orcid.org/0000-0001-5127-5459> a <schema:Person> .\r\n",
      "\r\n",
      "<http://orcid.org/0000-0003-4874-2874> a <schema:Person> .\r\n",
      "\r\n",
      "<https://bio.tools/bwa/op_2/in_2> a <http://bioschemas.org/types/FormalParameter> ;\r\n",
      "    ns1:identifier \"http://edamontology.org/data_2044\" ;\r\n",
      "    ns1:name \"Sequence\" ;\r\n",
      "    ns1:sameAs \"http://edamontology.org/data_2044\" .\r\n",
      "\r\n",
      "<https://bio.tools/bwa/op_3/in_2> a <http://bioschemas.org/types/FormalParameter> ;\r\n",
      "    ns1:identifier \"http://edamontology.org/data_2044\" ;\r\n",
      "    ns1:name \"Sequence\" ;\r\n",
      "    ns1:sameAs \"http://edamontology.org/data_2044\" .\r\n",
      "\r\n",
      "<https://bio.tools/bwa/op_4/in_2> a <http://bioschemas.org/types/FormalParameter> ;\r\n",
      "    ns1:identifier \"http://edamontology.org/data_2044\" ;\r\n",
      "    ns1:name \"Sequence\" ;\r\n",
      "    ns1:sameAs \"http://edamontology.org/data_2044\" .\r\n",
      "\r\n",
      "<https://bio.tools/bwa/op_4/in_3> a <http://bioschemas.org/types/FormalParameter> ;\r\n",
      "    ns1:identifier \"http://edamontology.org/data_2044\" ;\r\n",
      "    ns1:name \"Sequence\" ;\r\n",
      "    ns1:sameAs \"http://edamontology.org/data_2044\" .\r\n",
      "\r\n",
      "<https://bio.tools/bwa/op_5/in_2> a <http://bioschemas.org/types/FormalParameter> ;\r\n",
      "    ns1:identifier \"http://edamontology.org/data_2044\" ;\r\n",
      "    ns1:name \"Sequence\" ;\r\n",
      "    ns1:sameAs \"http://edamontology.org/data_2044\" .\r\n",
      "\r\n",
      "<https://bio.tools/bwa/op_5/in_3> a <http://bioschemas.org/types/FormalParameter> ;\r\n",
      "    ns1:identifier \"http://edamontology.org/data_2044\" ;\r\n",
      "    ns1:name \"Sequence\" ;\r\n",
      "    ns1:sameAs \"http://edamontology.org/data_2044\" .\r\n",
      "\r\n",
      "<https://bio.tools/bwa/op_6/in_2> a <http://bioschemas.org/types/FormalParameter> ;\r\n",
      "    ns1:identifier \"http://edamontology.org/data_2044\" ;\r\n",
      "    ns1:name \"Sequence\" ;\r\n",
      "    ns1:sameAs \"http://edamontology.org/data_2044\" .\r\n",
      "\r\n",
      "<https://bio.tools/jaspar/op_1/in_2> a <http://bioschemas.org/types/FormalParameter> ;\r\n",
      "    ns1:identifier \"http://edamontology.org/data_1077\" ;\r\n",
      "    ns1:name \"Transcription factor identifier\" ;\r\n",
      "    ns1:sameAs \"http://edamontology.org/data_1077\" .\r\n",
      "\r\n",
      "<https://bio.tools/jaspar/op_1/in_3> a <http://bioschemas.org/types/FormalParameter> ;\r\n",
      "    ns1:identifier \"http://edamontology.org/data_1868\" ;\r\n",
      "    ns1:name \"Taxon\" ;\r\n",
      "    ns1:sameAs \"http://edamontology.org/data_1868\" .\r\n",
      "\r\n",
      "<https://bio.tools/jaspar/op_1/in_4> a <http://bioschemas.org/types/FormalParameter> ;\r\n",
      "    ns1:identifier \"http://edamontology.org/data_2732\" ;\r\n",
      "    ns1:name \"Family name\" ;\r\n",
      "    ns1:sameAs \"http://edamontology.org/data_2732\" .\r\n",
      "\r\n",
      "<https://bio.tools/jaspar/op_1/in_5> a <http://bioschemas.org/types/FormalParameter> ;\r\n",
      "    ns1:identifier \"http://edamontology.org/data_1045\" ;\r\n",
      "    ns1:name \"Species name\" ;\r\n",
      "    ns1:sameAs \"http://edamontology.org/data_1045\" .\r\n",
      "\r\n",
      "<https://bio.tools/jaspar/op_1/in_6> a <http://bioschemas.org/types/FormalParameter> ;\r\n",
      "    ns1:identifier \"http://edamontology.org/data_2291\" ;\r\n",
      "    ns1:name \"UniProt ID\" ;\r\n",
      "    ns1:sameAs \"http://edamontology.org/data_2291\" .\r\n",
      "\r\n",
      "<https://bio.tools/jaspar/op_1/in_7> a <http://bioschemas.org/types/FormalParameter> ;\r\n",
      "    ns1:identifier \"http://edamontology.org/data_3671\" ;\r\n",
      "    ns1:name \"Text\" ;\r\n",
      "    ns1:sameAs \"http://edamontology.org/data_3671\" .\r\n",
      "\r\n",
      "<https://bio.tools/jaspar/op_2/in_2> a <http://bioschemas.org/types/FormalParameter> ;\r\n",
      "    ns1:identifier \"http://edamontology.org/data_1119\" ;\r\n",
      "    ns1:name \"JASPAR profile ID\" ;\r\n",
      "    ns1:sameAs \"http://edamontology.org/data_1119\" .\r\n",
      "\r\n",
      "<https://doi.org/10.1016/j.ygeno.2017.03.001> a ns1:CreativeWork .\r\n",
      "\r\n",
      "<https://doi.org/10.1093/bioinformatics/btp324> a ns1:CreativeWork .\r\n",
      "\r\n",
      "<https://doi.org/10.1093/bioinformatics/btp698> a ns1:CreativeWork .\r\n",
      "\r\n",
      "<https://doi.org/10.1093/bioinformatics/btu146> a ns1:CreativeWork .\r\n",
      "\r\n",
      "<https://doi.org/10.1093/bioinformatics/btx804> a ns1:CreativeWork .\r\n",
      "\r\n",
      "<https://doi.org/10.1093/nar/gkh012> a ns1:CreativeWork .\r\n",
      "\r\n",
      "<https://doi.org/10.1093/nar/gkj115> a ns1:CreativeWork .\r\n",
      "\r\n",
      "<https://doi.org/10.1093/nar/gkm955> a ns1:CreativeWork .\r\n",
      "\r\n",
      "<https://doi.org/10.1093/nar/gkp950> a ns1:CreativeWork .\r\n",
      "\r\n",
      "<https://doi.org/10.1093/nar/gkt997> a ns1:CreativeWork .\r\n",
      "\r\n",
      "<https://doi.org/10.1093/nar/gkv1176> a ns1:CreativeWork .\r\n",
      "\r\n",
      "<https://doi.org/10.1093/nar/gkx1126> a ns1:CreativeWork .\r\n",
      "\r\n",
      "<https://doi.org/10.1186/1471-2105-14-184> a ns1:CreativeWork .\r\n",
      "\r\n",
      "<https://doi.org/10.1186/1471-2164-15-264> a ns1:CreativeWork .\r\n",
      "\r\n",
      "<https://github.com/diwis/PaperRanking> a ns1:SoftwareSourceCode .\r\n",
      "\r\n",
      "<https://orcid.org/0000-0002-0336-9669> a ns1:Person ;\r\n",
      "    ns1:email \"tim.heitland@awi.de\" ;\r\n",
      "    ns1:familyName \"Heitland\" ;\r\n",
      "    ns1:givenName \"Tim\" ;\r\n",
      "    ns1:identifier \"https://orcid.org/0000-0002-0336-9669\" ;\r\n",
      "    ns1:name \"Tim Heitland\" .\r\n",
      "\r\n",
      "<https://orcid.org/0000-0002-3154-6212> a ns1:Person ;\r\n",
      "    ns1:affiliation \"Athena Research Center\" ;\r\n",
      "    ns1:name \"Danae Pla Karidi\" .\r\n",
      "\r\n",
      "<https://orcid.org/0000-0002-5002-7901> a ns1:Person ;\r\n",
      "    ns1:affiliation \"Athena Research Center\" ;\r\n",
      "    ns1:name \"Theodore Dalamagas\" .\r\n",
      "\r\n",
      "<https://orcid.org/0000-0002-9786-4010> a ns1:Person ;\r\n",
      "    ns1:email \"christine.wesche@awi.de\" ;\r\n",
      "    ns1:familyName \"Wesche\" ;\r\n",
      "    ns1:givenName \"Christine\" ;\r\n",
      "    ns1:identifier \"https://orcid.org/0000-0002-9786-4010\" ;\r\n",
      "    ns1:name \"Christine Wesche\" ;\r\n",
      "    ns1:url <http://www.awi.de/en/about-us/organisation/staff/christine-wesche.html> .\r\n",
      "\r\n",
      "<https://orcid.org/0000-0003-0555-4128> a ns1:Person ;\r\n",
      "    ns1:affiliation \"Athena Research Center\" ;\r\n",
      "    ns1:name \"Thanasis Vergoulis\" .\r\n",
      "\r\n",
      "<https://orcid.org/0000-0003-1714-5225> a ns1:Person ;\r\n",
      "    ns1:affiliation \"Athena Research Center\" ;\r\n",
      "    ns1:name \"Serafeim Chatzopoulos\" .\r\n",
      "\r\n",
      "<https://orcid.org/0000-0003-2146-3795> a ns1:Person ;\r\n",
      "    ns1:affiliation \"Athena Research Center\" ;\r\n",
      "    ns1:name \"Ilias Kanellos\" .\r\n",
      "\r\n",
      "<https://pages.semanticscholar.org/coronavirus-research> a ns1:Dataset .\r\n",
      "\r\n",
      "<https://bio.tools/bwa/op_1/in_1> a <http://bioschemas.org/types/FormalParameter> ;\r\n",
      "    ns1:identifier \"http://edamontology.org/data_2044\",\r\n",
      "        \"http://edamontology.org/data_3210\" ;\r\n",
      "    ns1:name \"Genome index\",\r\n",
      "        \"Sequence\" ;\r\n",
      "    ns1:sameAs \"http://edamontology.org/data_2044\",\r\n",
      "        \"http://edamontology.org/data_3210\" .\r\n",
      "\r\n",
      "<https://bio.tools/bwa/op_2/in_1> a <http://bioschemas.org/types/FormalParameter> ;\r\n",
      "    ns1:identifier \"http://edamontology.org/data_1916\",\r\n",
      "        \"http://edamontology.org/data_3210\" ;\r\n",
      "    ns1:name \"Alignment\",\r\n",
      "        \"Genome index\" ;\r\n",
      "    ns1:sameAs \"http://edamontology.org/data_1916\",\r\n",
      "        \"http://edamontology.org/data_3210\" .\r\n",
      "\r\n",
      "<https://bio.tools/bwa/op_3/in_1> a <http://bioschemas.org/types/FormalParameter> ;\r\n",
      "    ns1:identifier \"http://edamontology.org/data_2012\",\r\n",
      "        \"http://edamontology.org/data_2044\" ;\r\n",
      "    ns1:name \"Sequence\",\r",
      "\r\n",
      "        \"Sequence coordinates\" ;\r\n",
      "    ns1:sameAs \"http://edamontology.org/data_2012\",\r\n",
      "        \"http://edamontology.org/data_2044\" .\r\n",
      "\r\n",
      "<https://bio.tools/bwa/op_4/in_1> a <http://bioschemas.org/types/FormalParameter> ;\r\n",
      "    ns1:identifier \"http://edamontology.org/data_0863\",\r\n",
      "        \"http://edamontology.org/data_2044\" ;\r\n",
      "    ns1:name \"Sequence\",\r\n",
      "        \"Sequence alignment\" ;\r\n",
      "    ns1:sameAs \"http://edamontology.org/data_0863\",\r\n",
      "        \"http://edamontology.org/data_2044\" .\r\n",
      "\r\n",
      "<https://bio.tools/bwa/op_5/in_1> a <http://bioschemas.org/types/FormalParameter> ;\r\n",
      "    ns1:identifier \"http://edamontology.org/data_0863\",\r\n",
      "        \"http://edamontology.org/data_2044\" ;\r\n",
      "    ns1:name \"Sequence\",\r\n",
      "        \"Sequence alignment\" ;\r\n",
      "    ns1:sameAs \"http://edamontology.org/data_0863\",\r\n",
      "        \"http://edamontology.org/data_2044\" .\r\n",
      "\r\n",
      "<https://bio.tools/bwa/op_6/in_1> a <http://bioschemas.org/types/FormalParameter> ;\r\n",
      "    ns1:identifier \"http://edamontology.org/data_0863\",\r\n",
      "        \"http://edamontology.org/data_2044\" ;\r\n",
      "    ns1:name \"Sequence\",\r\n",
      "        \"Sequence alignment\" ;\r\n",
      "    ns1:sameAs \"http://edamontology.org/data_0863\",\r\n",
      "        \"http://edamontology.org/data_2044\" .\r\n",
      "\r\n",
      "<https://bio.tools/jaspar/op_1/in_1> a <http://bioschemas.org/types/FormalParameter> ;\r\n",
      "    ns1:identifier \"http://edamontology.org/data_1119\",\r\n",
      "        \"http://edamontology.org/data_2755\" ;\r\n",
      "    ns1:name \"JASPAR profile ID\",\r\n",
      "        \"Transcription factor name\" ;\r\n",
      "    ns1:sameAs \"http://edamontology.org/data_1119\",\r\n",
      "        \"http://edamontology.org/data_2755\" .\r\n",
      "\r\n",
      "<https://bio.tools/jaspar/op_2/in_1> a <http://bioschemas.org/types/FormalParameter> ;\r\n",
      "    ns1:identifier \"http://edamontology.org/data_1119\",\r\n",
      "        \"http://edamontology.org/data_3494\" ;\r\n",
      "    ns1:name \"DNA sequence\",\r\n",
      "        \"JASPAR profile ID\" ;\r\n",
      "    ns1:sameAs \"http://edamontology.org/data_1119\",\r\n",
      "        \"http://edamontology.org/data_3494\" .\r\n",
      "\r\n",
      "<https://orcid.org/0000-0002-9022-3069> a ns1:Person ;\r\n",
      "    ns1:email \"mia.wege@gmail.com\" ;\r\n",
      "    ns1:familyName \"Wege\" ;\r\n",
      "    ns1:givenName \"Mia\" ;\r\n",
      "    ns1:identifier \"https://orcid.org/0000-0002-9022-3069\" ;\r\n",
      "    ns1:name \"Mia Wege\" .\r\n",
      "\r\n",
      "<https://orcid.org/0000-0002-9357-9853> a ns1:Person ;\r\n",
      "    ns1:email \"hartmut.hellmer@awi.de\" ;\r\n",
      "    ns1:familyName \"Hellmer\" ;\r\n",
      "    ns1:givenName \"Hartmut H\" ;\r\n",
      "    ns1:identifier \"https://orcid.org/0000-0002-9357-9853\" ;\r\n",
      "    ns1:name \"Hartmut H Hellmer\" ;\r\n",
      "    ns1:url <http://www.awi.de/en/about-us/organisation/staff/hartmut-hellmer.html> .\r\n",
      "\r\n",
      "<https://orcid.org/0000-0003-3060-0317> a ns1:Person ;\r\n",
      "    ns1:email \"Elin.Darelius@uib.no\" ;\r\n",
      "    ns1:familyName \"Darelius\" ;\r\n",
      "    ns1:givenName \"Elin\" ;\r\n",
      "    ns1:identifier \"https://orcid.org/0000-0003-3060-0317\" ;\r\n",
      "    ns1:name \"Elin Darelius\" .\r\n",
      "\r\n",
      "<https://doi.org/10.48433/BzPM_0758_2021> a ns1:CreativeWork,\r\n",
      "        ns1:PublicationIssue ;\r\n",
      "    ns1:creator [ a ns1:Person ;\r\n",
      "            ns1:email \"Tanja.Fromm@awi.de\" ;\r\n",
      "            ns1:familyName \"Fromm\" ;\r\n",
      "            ns1:givenName \"Tanja\" ;\r\n",
      "            ns1:name \"Tanja Fromm\" ],\r\n",
      "        [ a ns1:Person ;\r\n",
      "            ns1:familyName \"Oberdieck\" ;\r\n",
      "            ns1:givenName \"Constance\" ;\r\n",
      "            ns1:name \"Constance Oberdieck\" ],\r\n",
      "        [ a ns1:Person ;\r\n",
      "            ns1:familyName \"Oberdieck\" ;\r\n",
      "            ns1:givenName \"Constance\" ;\r\n",
      "            ns1:name \"Constance Oberdieck\" ],\r\n",
      "        [ a ns1:Person ;\r\n",
      "            ns1:email \"Tanja.Fromm@awi.de\" ;\r\n",
      "            ns1:familyName \"Fromm\" ;\r\n",
      "            ns1:givenName \"Tanja\" ;\r\n",
      "            ns1:name \"Tanja Fromm\" ],\r\n",
      "        <https://orcid.org/0000-0002-0336-9669>,\r\n",
      "        <https://orcid.org/0000-0002-9786-4010> ;\r\n",
      "    ns1:datePublished \"2021\"^^ns1:Date ;\r\n",
      "    ns1:identifier \"https://doi.org/10.48433/BzPM_0758_2021\" ;\r\n",
      "    ns1:isPartOf [ a ns1:CreativeWorkSeries ;\r\n",
      "            ns1:name \"Berichte zur Polar- und Meeresforschung = Reports on Polar and Marine Research\" ],\r\n",
      "        [ a ns1:CreativeWorkSeries ;\r\n",
      "            ns1:name \"Berichte zur Polar- und Meeresforschung = Reports on Polar and Marine Research\" ] ;\r\n",
      "    ns1:issueNumber \"758\" ;\r\n",
      "    ns1:name \"Expeditions to Antarctica: ANT-Land 2020/21 NEUMAYER STATION III, Kohnen Station, Flight Operations and Field Campaigns\" ;\r\n",
      "    ns1:pagination \"99 pp\" ;\r\n",
      "    ns1:url <https://doi.org/10.48433/BzPM_0758_2021> .\r\n",
      "\r\n",
      "[] a ns1:Dataset ;\r\n",
      "    ns1:abstract \"Weddell seals (Leptonychotes weddellii) were instrumented during expedition PS124 (2021) for the purpose of deployments with satellite transmitters. The deployments (n=10) were located in the Filchner-Outflow-System (FOS), southern Weddell Sea. The seals (7 female, 3 male) were captured along the ships' tracks during ahead flights of the board helicopter and immobilised and instrumented on pack ice with Argos-linked (CLS, Toulouse, France) satellite-relayed dive loggers combined with CTD's (CTD-SRDLs, Sea Mammal Research Unit, UK) to investigate their ranging and foraging behaviour in the context of oceanographic features. Six of the ten CTD-SRDLs provided data. The average lifetime of PTTs was 124.7 days ( MIN 2; MAX 233). The entire workflow and data processing routines are outlined in a data processing report.\" ;\r\n",
      "    ns1:citation <https://doi.org/10.48433/BzPM_0758_2021> ;\r\n",
      "    ns1:conditionsOfAccess \"access rights needed\" ;\r\n",
      "    ns1:creator [ a ns1:Person ;\r\n",
      "            ns1:email \"horst.bornemann@awi.de\" ;\r\n",
      "            ns1:familyName \"Bornemann\" ;\r\n",
      "            ns1:givenName \"Horst\" ;\r\n",
      "            ns1:name \"Horst Bornemann\" ;\r\n",
      "            ns1:url <http://www.awi.de/en/about-us/organisation/staff/horst-bornemann.html> ],\r\n",
      "        <https://orcid.org/0000-0002-9022-3069>,\r\n",
      "        <https://orcid.org/0000-0002-9357-9853>,\r\n",
      "        <https://orcid.org/0000-0003-3060-0317> ;\r\n",
      "    ns1:datePublished \"2022-04-05\"^^ns1:Date ;\r\n",
      "    ns1:description \"Weddell seals (Leptonychotes weddellii) were instrumented during expedition PS124 (2021) for the purpose of deployments with satellite transmitters. The deployments (n=10) were located in the Filchner-Outflow-System (FOS), southern Weddell Sea. The seals (7 female, 3 male) were captured along the ships' tracks during ahead flights of the board helicopter and immobilised and instrumented on pack ice with Argos-linked (CLS, Toulouse, France) satellite-relayed dive loggers combined with CTD's (CTD-SRDLs, Sea Mammal Research Unit, UK) to investigate their ranging and foraging behaviour in the context of oceanographic features. Six of the ten CTD-SRDLs provided data. The average lifetime of PTTs was 124.7 days ( MIN 2; MAX 233). The entire workflow and data processing routines are outlined in a data processing report.\" ;\r\n",
      "    ns1:distribution [ a ns1:DataDownload ;\r\n",
      "            ns1:contentUrl <https://doi.pangaea.de/10.1594/PANGAEA.942997?format=zip> ;\r\n",
      "            ns1:encodingFormat \"application/zip\" ] ;\r\n",
      "    ns1:inLanguage \"en\" ;\r\n",
      "    ns1:includedInDataCatalog [ a ns1:DataCatalog ;\r\n",
      "            ns1:disambiguatingDescription \"Data Publisher for Earth & Environmental Science\" ;\r\n",
      "            ns1:name \"PANGAEA\" ;\r\n",
      "            ns1:url <https://www.pangaea.de/> ] ;\r\n",
      "    ns1:isAccessibleForFree false ;\r\n",
      "    ns1:keywords \"Leptonychotes weddellii; satellite tracking; Weddell seal\" ;\r\n",
      "    ns1:name \"Weddell seal tracking and diving data from expedition FIL2021\" ;\r\n",
      "    ns1:publisher [ a ns1:Organization ;\r\n",
      "            ns1:disambiguatingDescription \"Data Publisher for Earth & Environmental Science\" ;\r\n",
      "            ns1:name \"PANGAEA\" ;\r\n",
      "            ns1:url <https://www.pangaea.de/> ] ;\r\n",
      "    ns1:size [ a ns1:QuantitativeValue ;\r\n",
      "            ns1:unitText \"datasets\" ;\r\n",
      "            ns1:value 1.9e+01 ] ;\r\n",
      "    ns1:spatialCoverage [ a ns1:Place ;\r\n",
      "            ns1:geo [ a ns1:GeoShape ;\r\n",
      "                    ns1:box \"-77.726 -44.27600000000001 -74.074 -25.054\" ] ] ;\r\n",
      "    ns1:temporalCoverage \"2021-02-14T23:20:00/2021-10-09T07:20:00\" ;\r\n",
      "    ns1:url <https://doi.pangaea.de/10.1594/PANGAEA.942997> .\r\n",
      "\r\n",
      "[] a ns1:Dataset ;\r\n",
      "    ns1:abstract \"Weddell seals (Leptonychotes weddellii) were instrumented during expedition PS124 (2021) for the purpose of deployments with satellite transmitters. The deployments (n=10) were located in the Filchner-Outflow-System (FOS), southern Weddell Sea. The seals (7 female, 3 male) were captured along the ships' tracks during ahead flights of the board helicopter and immobilised and instrumented on pack ice with Argos-linked (CLS, Toulouse, France) satellite-relayed dive loggers combined with CTD's (CTD-SRDLs, Sea Mammal Research Unit, UK) to investigate their ranging and foraging behaviour in the context of oceanographic features. Six of the ten CTD-SRDLs provided data. The average lifetime of PTTs was 124.7 days ( MIN 2; MAX 233). The entire workflow and data processing routines are outlined in a data processing report.\" ;\r\n",
      "    ns1:citation <https://doi.org/10.48433/BzPM_0758_2021> ;\r\n",
      "    ns1:conditionsOfAccess \"access rights needed\" ;\r\n",
      "    ns1:creator [ a ns1:Person ;\r\n",
      "            ns1:email \"horst.bornemann@awi.de\" ;\r\n",
      "            ns1:familyName \"Bornemann\" ;\r\n",
      "            ns1:givenName \"Horst\" ;\r\n",
      "            ns1:name \"Horst Bornemann\" ;\r\n",
      "            ns1:url <http://www.awi.de/en/about-us/organisation/staff/horst-bornemann.html> ],\r\n",
      "        <https://orcid.org/0000-0002-9022-3069>,\r\n",
      "        <https://orcid.org/0000-0002-9357-9853>,\r\n",
      "        <https://orcid.org/0000-0003-3060-0317> ;\r\n",
      "    ns1:datePublished \"2022-04-05\"^^ns1:Date ;\r\n",
      "    ns1:description \"Weddell seals (Leptonychotes weddellii) were instrumented during expedition PS124 (2021) for the purpose of deployments with satellite transmitters. The deployments (n=10) were located in the Filchner-Outflow-System (FOS), southern Weddell Sea. The seals (7 female, 3 male) were captured along the ships' tracks during ahead flights of the board helicopter and immobilised and instrumented on pack ice with Argos-linked (CLS, Toulouse, France) satellite-relayed dive loggers combined with CTD's (CTD-SRDLs, Sea Mammal Research Unit, UK) to investigate their ranging and foraging behaviour in the context of oceanographic features. Six of the ten CTD-SRDLs provided data. The average lifetime of PTTs was 124.7 days ( MIN 2; MAX 233). The entire workflow and data processing routines are outlined in a data processing report.\" ;\r\n",
      "    ns1:distribution [ a ns1:DataDownload ;\r\n",
      "            ns1:contentUrl <https://doi.pangaea.de/10.1594/PANGAEA.942997?format=zip> ;\r\n",
      "            ns1:encodingFormat \"application/zip\" ] ;\r\n",
      "    ns1:inLanguage \"en\" ;\r\n",
      "    ns1:includedInDataCatalog [ a ns1:DataCatalog ;\r\n",
      "            ns1:disambiguatingDescription \"Data Publisher for Earth & Environmental Science\" ;\r\n",
      "            ns1:name \"PANGAEA\" ;\r\n",
      "            ns1:url <https://www.pangaea.de/> ] ;\r\n",
      "    ns1:isAccessibleForFree false ;\r\n",
      "    ns1:keywords \"Leptonychotes weddellii; satellite tracking; Weddell seal\" ;\r\n",
      "    ns1:name \"Weddell seal tracking and diving data from expedition FIL2021\" ;\r\n",
      "    ns1:publisher [ a ns1:Organization ;\r\n",
      "            ns1:disambiguatingDescription \"Data Publisher for Earth & Environmental Science\" ;\r\n",
      "            ns1:name \"PANGAEA\" ;\r\n",
      "            ns1:url <https://www.pangaea.de/> ] ;\r\n",
      "    ns1:size [ a ns1:QuantitativeValue ;\r\n",
      "            ns1:unitText \"datasets\" ;\r\n",
      "            ns1:value 1.9e+01 ] ;\r\n",
      "    ns1:spatialCoverage [ a ns1:Place ;\r\n",
      "            ns1:geo [ a ns1:GeoShape ;\r\n",
      "                    ns1:box \"-77.726 -44.27600000000001 -74.074 -25.054\" ] ] ;\r\n",
      "    ns1:temporalCoverage \"2021-02-14T23:20:00/2021-10-09T07:20:00\" ;\r\n",
      "    ns1:url <https://doi.pangaea.de/10.1594/PANGAEA.942997> .\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!cat scrapped_bioschemas.ttl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 3. Display Bioschemas classes and properties counts."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Used classes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>c</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>http://bioschemas.org/types/FormalParameter</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>http://schema.org/Person</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>http://schema.org/CreativeWork</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>http://schema.org/DataDownload</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>http://schema.org/Dataset</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>http://schema.org/SoftwareApplication</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>schema:Person</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>http://schema.org/QuantitativeValue</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>http://schema.org/Place</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>http://schema.org/CreativeWorkSeries</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>http://schema.org/Organization</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>http://schema.org/DataCatalog</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>http://schema.org/GeoShape</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>http://schema.org/PublicationIssue</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>http://schema.org/SoftwareSourceCode</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              c count\n",
       "0   http://bioschemas.org/types/FormalParameter    22\n",
       "1                      http://schema.org/Person    16\n",
       "2                http://schema.org/CreativeWork    15\n",
       "3                http://schema.org/DataDownload     7\n",
       "4                     http://schema.org/Dataset     4\n",
       "5         http://schema.org/SoftwareApplication     2\n",
       "6                                 schema:Person     2\n",
       "7           http://schema.org/QuantitativeValue     2\n",
       "8                       http://schema.org/Place     2\n",
       "9          http://schema.org/CreativeWorkSeries     2\n",
       "10               http://schema.org/Organization     2\n",
       "11                http://schema.org/DataCatalog     2\n",
       "12                   http://schema.org/GeoShape     2\n",
       "13           http://schema.org/PublicationIssue     1\n",
       "14         http://schema.org/SoftwareSourceCode     1"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def get_dataframe_from_query_results(res):    \n",
    "    return pd.DataFrame(res.bindings)\n",
    "\n",
    "### display used classes\n",
    "classes_counts = \"\"\"\n",
    "SELECT ?c (count(?c) as ?count) WHERE {\n",
    "    ?s rdf:type ?c .\n",
    "} \n",
    "GROUP BY ?c\n",
    "ORDER BY DESC(?count)\n",
    "\"\"\"\n",
    "\n",
    "res = KG.query(classes_counts)\n",
    "print()\n",
    "print(\"Used classes\")\n",
    "df = get_dataframe_from_query_results(res)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Used properties\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "      <th>p</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>82</td>\n",
       "      <td>http://www.w3.org/1999/02/22-rdf-syntax-ns#type</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>56</td>\n",
       "      <td>http://schema.org/name</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>37</td>\n",
       "      <td>http://schema.org/identifier</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>34</td>\n",
       "      <td>http://schema.org/citation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>30</td>\n",
       "      <td>http://schema.org/sameAs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>22</td>\n",
       "      <td>http://bioschemas.org/input</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>19</td>\n",
       "      <td>http://schema.org/creator</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>12</td>\n",
       "      <td>http://schema.org/url</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>11</td>\n",
       "      <td>http://schema.org/givenName</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>11</td>\n",
       "      <td>http://schema.org/familyName</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>9</td>\n",
       "      <td>http://schema.org/email</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>8</td>\n",
       "      <td>http://bioschemas.org/output</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>7</td>\n",
       "      <td>http://schema.org/contentUrl</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>7</td>\n",
       "      <td>http://schema.org/encodingFormat</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>7</td>\n",
       "      <td>http://schema.org/applicationSubCategory</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>7</td>\n",
       "      <td>http://schema.org/distribution</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>6</td>\n",
       "      <td>http://schema.org/keywords</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>6</td>\n",
       "      <td>http://schema.org/featureList</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>5</td>\n",
       "      <td>http://schema.org/affiliation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>5</td>\n",
       "      <td>https://bio.tools/ontology/primaryContact</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>4</td>\n",
       "      <td>http://schema.org/datePublished</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>4</td>\n",
       "      <td>http://schema.org/disambiguatingDescription</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>4</td>\n",
       "      <td>http://schema.org/softwareHelp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>3</td>\n",
       "      <td>http://schema.org/description</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>3</td>\n",
       "      <td>http://schema.org/isAccessibleForFree</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>2</td>\n",
       "      <td>http://schema.org/value</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>2</td>\n",
       "      <td>http://schema.org/unitText</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>2</td>\n",
       "      <td>http://schema.org/size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>2</td>\n",
       "      <td>http://schema.org/includedInDataCatalog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>2</td>\n",
       "      <td>http://schema.org/geo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>2</td>\n",
       "      <td>http://schema.org/isPartOf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>2</td>\n",
       "      <td>http://schema.org/inLanguage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>2</td>\n",
       "      <td>http://schema.org/temporalCoverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>2</td>\n",
       "      <td>http://schema.org/box</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>2</td>\n",
       "      <td>http://purl.org/dc/terms/conformsTo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>2</td>\n",
       "      <td>http://schema.org/spatialCoverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>2</td>\n",
       "      <td>http://ogp.me/ns#image</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>2</td>\n",
       "      <td>http://schema.org/conditionsOfAccess</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>2</td>\n",
       "      <td>http://schema.org/abstract</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>2</td>\n",
       "      <td>http://ogp.me/ns#title</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>2</td>\n",
       "      <td>http://ogp.me/ns#description</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>2</td>\n",
       "      <td>http://schema.org/publisher</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>1</td>\n",
       "      <td>http://ogp.me/ns#site_name</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>1</td>\n",
       "      <td>http://schema.org/issueNumber</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>1</td>\n",
       "      <td>http://schema.org/contributor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>1</td>\n",
       "      <td>http://ogp.me/ns#url</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>1</td>\n",
       "      <td>http://schema.org/version</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>1</td>\n",
       "      <td>http://schema.org/pagination</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>1</td>\n",
       "      <td>http://schema.org/license</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   count                                                p\n",
       "0     82  http://www.w3.org/1999/02/22-rdf-syntax-ns#type\n",
       "1     56                           http://schema.org/name\n",
       "2     37                     http://schema.org/identifier\n",
       "3     34                       http://schema.org/citation\n",
       "4     30                         http://schema.org/sameAs\n",
       "5     22                      http://bioschemas.org/input\n",
       "6     19                        http://schema.org/creator\n",
       "7     12                            http://schema.org/url\n",
       "8     11                      http://schema.org/givenName\n",
       "9     11                     http://schema.org/familyName\n",
       "10     9                          http://schema.org/email\n",
       "11     8                     http://bioschemas.org/output\n",
       "12     7                     http://schema.org/contentUrl\n",
       "13     7                 http://schema.org/encodingFormat\n",
       "14     7         http://schema.org/applicationSubCategory\n",
       "15     7                   http://schema.org/distribution\n",
       "16     6                       http://schema.org/keywords\n",
       "17     6                    http://schema.org/featureList\n",
       "18     5                    http://schema.org/affiliation\n",
       "19     5        https://bio.tools/ontology/primaryContact\n",
       "20     4                  http://schema.org/datePublished\n",
       "21     4      http://schema.org/disambiguatingDescription\n",
       "22     4                   http://schema.org/softwareHelp\n",
       "23     3                    http://schema.org/description\n",
       "24     3            http://schema.org/isAccessibleForFree\n",
       "25     2                          http://schema.org/value\n",
       "26     2                       http://schema.org/unitText\n",
       "27     2                           http://schema.org/size\n",
       "28     2          http://schema.org/includedInDataCatalog\n",
       "29     2                            http://schema.org/geo\n",
       "30     2                       http://schema.org/isPartOf\n",
       "31     2                     http://schema.org/inLanguage\n",
       "32     2               http://schema.org/temporalCoverage\n",
       "33     2                            http://schema.org/box\n",
       "34     2              http://purl.org/dc/terms/conformsTo\n",
       "35     2                http://schema.org/spatialCoverage\n",
       "36     2                           http://ogp.me/ns#image\n",
       "37     2             http://schema.org/conditionsOfAccess\n",
       "38     2                       http://schema.org/abstract\n",
       "39     2                           http://ogp.me/ns#title\n",
       "40     2                     http://ogp.me/ns#description\n",
       "41     2                      http://schema.org/publisher\n",
       "42     1                       http://ogp.me/ns#site_name\n",
       "43     1                    http://schema.org/issueNumber\n",
       "44     1                    http://schema.org/contributor\n",
       "45     1                             http://ogp.me/ns#url\n",
       "46     1                        http://schema.org/version\n",
       "47     1                     http://schema.org/pagination\n",
       "48     1                        http://schema.org/license"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### display used properties\n",
    "property_counts = \"\"\"\n",
    "SELECT ?p (count(?p) as ?count) WHERE {\n",
    "    ?s ?p ?o .\n",
    "} \n",
    "GROUP BY ?p\n",
    "ORDER BY DESC(?count)\n",
    "\"\"\"\n",
    "\n",
    "res = KG.query(property_counts)\n",
    "print()\n",
    "print(\"Used properties\")\n",
    "df = get_dataframe_from_query_results(res)\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Visualization\n",
    "Have a look to Seaborn and Bokeh libraries\n",
    "https://seaborn.pydata.org\n",
    "https://bokeh.org"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "interpreter": {
   "hash": "e202b6cb132b9357ed023fecdbc194f9a7f520726b3b39fdf8a2279e78a83e12"
  },
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
