{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 186,
   "id": "7dd4a930",
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "sys.path.append(\"../\")\n",
    "\n",
    "from rdflib import ConjunctiveGraph, URIRef\n",
    "import requests\n",
    "from requests.exceptions import InvalidURL\n",
    "import random\n",
    "import json\n",
    "from tqdm.notebook import tqdm\n",
    "import pandas as pd\n",
    "from metrics.WebResource import WebResource"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "id": "d035e081",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_live_deploys_urls():\n",
    "    results = []\n",
    "    live_deploys_remote_file = \"https://raw.githubusercontent.com/BioSchemas/bioschemas.github.io/master/_data/live_deployments.json\"\n",
    "    res = requests.get(live_deploys_remote_file)\n",
    "    live_deploys = res.json()\n",
    "    \n",
    "    for r in live_deploys[\"resources\"]:\n",
    "        for p in r[\"profiles\"]:\n",
    "            if \"exampleURL\" in p.keys():\n",
    "                results.append(p[\"exampleURL\"])\n",
    "            else : \n",
    "                print(f\"no example URL for profile {p['profileName']} and url {r['url']}\")\n",
    "    print(f\"found {len(set(results))} live deploys\")\n",
    "    return list(set(results))\n",
    "\n",
    "\n",
    "\n",
    "def test_ld_bioschemas_annot():\n",
    "        res = get_live_deploys_urls()\n",
    "        errors = []\n",
    "        results = []\n",
    "        \n",
    "        i = 0\n",
    "        for r in tqdm(res):\n",
    "        #for r in reversed(res):\n",
    "            row = {}\n",
    "            status_code = requests.head(r).status_code\n",
    "            print(r)\n",
    "            row['url'] = r\n",
    "            print(\"Status code: \" + str(status_code))\n",
    "            row['status code'] = str(status_code)\n",
    "            try :\n",
    "                kg = WebResource(r).get_rdf()\n",
    "                print(\"Triples: \" + str(len(kg)))\n",
    "                row['Retrieved triples'] = len(kg)\n",
    "                if not len(kg) > 0:\n",
    "                    print(f\"# Error with {r}\")\n",
    "                    row['Error'] = True\n",
    "                    errors.append(r)\n",
    "            except UnicodeDecodeError as err:\n",
    "                print(err)\n",
    "            results.append(row)\n",
    "\n",
    "        print(f\"{len(res)} tested URLS\")\n",
    "        print(f\"{len(errors)} failing URLS\")\n",
    "        print(errors)\n",
    "        df = pd.read_json(json.dumps(results))\n",
    "        return df\n",
    "    \n",
    "def retrieve_bioschemas(url):  \n",
    "    FC_get_md = \"https://fair-checker.france-bioinformatique.fr/api/inspect/get_rdf_metadata\"\n",
    "    kg = ConjunctiveGraph()\n",
    "    res = requests.get(url=FC_get_md, params={\"url\": url})\n",
    "    try:\n",
    "        kg.parse(data=res.text, format=\"json-ld\")\n",
    "    except Exception as e:\n",
    "            print(e)\n",
    "    print(f\"Loaded {len(kg)} RDF triples from {url}\")\n",
    "    return kg\n",
    "\n",
    "def check_bioschemas(url):  \n",
    "    #FC_bs_valid = \"https://fair-checker.france-bioinformatique.fr/api/inspect/bioschemas_validation\"\n",
    "    FC_bs_valid = \"https://fair-checker.france-bioinformatique.fr/api/inspect/bioschemas_validation_by_conformsto\"\n",
    "    res = requests.get(url=FC_bs_valid, params={\"url\": url})\n",
    "    #print(res.text)\n",
    "    #try:\n",
    "    #    print(json.dumps(res.json(), indent=True))\n",
    "    #except Exception as e:\n",
    "    #    print(e)\n",
    "    return res.json()\n",
    "\n",
    "def has_conforms_to(kg):\n",
    "    conformsTo = URIRef(\"http://purl.org/dc/terms/conformsTo\")\n",
    "    return (None, conformsTo, None) in kg"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ca494b2e",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 182,
   "id": "d2dbb0af",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "found 136 live deploys\n",
      "retrieved 136 live deploys\n"
     ]
    }
   ],
   "source": [
    "lds = get_live_deploys_urls()\n",
    "#lds = random.sample(lds,20)\n",
    "print(f\"retrieved {len(lds)} live deploys\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "id": "2c8803f6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "b33a1310f68f4bd7a9c8c841db4d34ef",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/136 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded 14 RDF triples from https://bridgedb.github.io/data/gene_database/\n",
      "Loaded 95 RDF triples from http://itsonedb.cloud.ba.infn.it/\n",
      "Loaded 124 RDF triples from https://www.sib.swiss/training/course/20230426_DOCK\n",
      "Loaded 0 RDF triples from https://proteinensemble.org/PED00014\n",
      "Loaded 25 RDF triples from https://bioregistry.io/registry/hgnc\n",
      "Loaded 373 RDF triples from https://humanmine.org/\n",
      "Loaded 17 RDF triples from https://modelarchive.org/\n",
      "Loaded 13 RDF triples from https://www.orpha.net/consor/cgi-bin/OC_Exp.php?Expert=141189&lng=en\n",
      "Loaded 0 RDF triples from https://deb-central.org/\n",
      "Loaded 34 RDF triples from https://bio.tools/blast\n",
      "Loaded 10 RDF triples from https://scholia.toolforge.org/chemical/Q18216\n",
      "Loaded 36 RDF triples from https://bmrb.io/data_library/summary/index.php?bmrbId=30309\n",
      "Loaded 26 RDF triples from https://bioschemas.org/tutorials/what_why_bioschemas\n",
      "Loaded 0 RDF triples from https://proteinensemble.org/\n",
      "Loaded 28 RDF triples from http://edgar.biocomp.unibo.it/\n",
      "Loaded 8 RDF triples from https://www.alliancegenome.org/gene/MGI:2442292\n",
      "Loaded 23 RDF triples from https://datacatalog.elixir-luxembourg.org/e/dataset/924dfe7a-71e8-11eb-bafe-3e22fbb3883f\n",
      "Loaded 71 RDF triples from https://www.metanetx.org/chem_info/MNXM680\n",
      "Loaded 32 RDF triples from https://nanocommons.github.io/tutorials/enteringData\n",
      "Loaded 69 RDF triples from https://zbmed.github.io/damalos/\n",
      "Loaded 48 RDF triples from https://swissmodel.expasy.org/\n",
      "Loaded 12 RDF triples from https://bridgedb.github.io/\n",
      "Loaded 16 RDF triples from https://www.birmingham.ac.uk/facilities/metabolomics-training-centre/courses/q-exactive.aspx\n",
      "Loaded 46 RDF triples from https://www.metanetx.org/\n",
      "Loaded 32 RDF triples from https://enanomapper.github.io/tutorials/Entering_and_analysing_nano_safety_data/readme.html\n",
      "Loaded 12 RDF triples from https://workflowhub.eu/\n",
      "Loaded 51 RDF triples from https://mint.bio.uniroma2.it/\n",
      "Loaded 12 RDF triples from https://bio.tools/covidmine\n",
      "Loaded 8 RDF triples from https://cancer.sanger.ac.uk/cosmic/\n",
      "Loaded 54 RDF triples from https://disprot.org/DP00086r026\n",
      "Loaded 77 RDF triples from https://www.covid19dataportal.org/\n",
      "Loaded 14 RDF triples from https://h2020-riskgone.github.io/riskgone-materials/identifiers_and_guidance.html\n",
      "Loaded 8 RDF triples from https://www.alliancegenome.org/gene/RGD:620474\n",
      "Loaded 49 RDF triples from https://workflowhub.eu/workflows/18\n",
      "Loaded 49 RDF triples from https://string-db.org/network/9606.ENSP00000379258\n",
      "Loaded 41 RDF triples from https://bgee.org/?page=species&amp;species_id=9606#expr-calls\n",
      "Loaded 15 RDF triples from https://prosite.expasy.org\n",
      "Loaded 1 RDF triples from http://www.ebi.ac.uk/pdbe/entry/pdb/4a7l\n",
      "Loaded 12 RDF triples from https://pathvisio.org/\n",
      "Loaded 52 RDF triples from https://fairsharing.org/\n",
      "Loaded 43 RDF triples from https://bar.biocomp.unibo.it/bar3/\n",
      "Loaded 36 RDF triples from https://pcddb.cryst.bbk.ac.uk/\n",
      "Loaded 19 RDF triples from https://www.ebi.ac.uk/biosamples/\n",
      "Loaded 13 RDF triples from https://massbank.eu/MassBank/RecordDisplay?id=LQB00001&dsn=RIKEN_IMS\n",
      "Loaded 40 RDF triples from https://pairedomicsdata.bioinformatics.nl/projects/d0f1add5-5b16-4f78-9b93-0d1d9c9bcb59.1\n",
      "Loaded 46 RDF triples from https://tess.elixir-europe.org/materials/hands-on-for-exome-sequencing-data-analysis-for-diagnosing-a-genetic-disease-tutorial\n",
      "Loaded 214 RDF triples from https://www.ebi.ac.uk/interpro/\n",
      "Loaded 43 RDF triples from https://cells.ebisc.org/BIHi006-D\n",
      "Loaded 73 RDF triples from https://rdmkit.elixir-europe.org/human_data.html\n",
      "Loaded 64 RDF triples from https://disprot.org/\n",
      "Loaded 37 RDF triples from https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/HGNC:37133\n",
      "Loaded 0 RDF triples from https://aip.fipapatients.org/\n",
      "Loaded 0 RDF triples from https://www.opalesurfcasting.net/connaissances/faune-et-flore/le-saumon-de-l-atlantique-salmo-salar.html\n",
      "Loaded 11 RDF triples from https://ippidb.pasteur.fr/\n",
      "Loaded 0 RDF triples from https://flymine.org/\n",
      "Loaded 36 RDF triples from https://tess.elixir-europe.org/events/introduction-to-r-for-biologists-online\n",
      "Loaded 8 RDF triples from https://zbmed.github.io/damalos/docs/2020.html\n",
      "Loaded 37 RDF triples from https://www.genenames.org/data/gene-symbol-report/#data-set\n",
      "Loaded 0 RDF triples from https://www.proteinatlas.org/\n",
      "Loaded 56 RDF triples from http://mvid-central.org/\n",
      "Loaded 763 RDF triples from https://disprot.org/DP00086\n",
      "Loaded 69 RDF triples from https://mobidb.org/\n",
      "Loaded 41 RDF triples from https://bgee.org/\n",
      "Loaded 49 RDF triples from https://www.chd7.org/\n",
      "Loaded 0 RDF triples from https://www.metanetx.org/ftp/\n",
      "Loaded 47 RDF triples from https://hpscreg.eu/cell-line/BIHi006-D\n",
      "Loaded 1 RDF triples from https://synbiohub.org/public/igem/BBa_K165005/1\n",
      "Loaded 37 RDF triples from https://ega-archive.org/\n",
      "Loaded 0 RDF triples from http://159.149.160.88/cscan\n",
      "Loaded 15 RDF triples from https://hamap.expasy.org/cgi-bin/unirule/unirule_browse.cgi\n",
      "Loaded 1 RDF triples from https://www.ebi.ac.uk/pdbe/entry/pdb/5hbi/protein/1\n",
      "Loaded 260 RDF triples from http://edgar.biocomp.unibo.it/cgi-bin/gene_disease_db/gene.py?gene=A2M\n",
      "Loaded 7 RDF triples from https://nanocommons.github.io/specifications/jrc/\n",
      "Loaded 85 RDF triples from https://www.gbif.org/dataset/c0cc29de-f49f-4b66-b4ec-c83afbb7101d\n",
      "Loaded 1 RDF triples from https://synbiohub.org/\n",
      "Loaded 28 RDF triples from https://pairedomicsdata.bioinformatics.nl/\n",
      "Loaded 7 RDF triples from https://www.guidetopharmacology.org/coronavirus.jsp\n",
      "Loaded 171 RDF triples from https://training.galaxyproject.org/training-material/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html\n",
      "Loaded 0 RDF triples from https://www.ebi.ac.uk/biosamples/samples/SAMEA491372\n",
      "Loaded 0 RDF triples from https://galaxyproject.org/events/\n",
      "Loaded 18 RDF triples from https://datacatalog.elixir-luxembourg.org/\n",
      "Loaded 7 RDF triples from https://scholia.toolforge.org/taxon/Q15978631\n",
      "Loaded 21 RDF triples from https://ega-archive.org/studies/EGAS00000000001\n",
      "Loaded 28 RDF triples from http://phenpath.biocomp.unibo.it/cgi-bin/phenpath/ppDB.py\n",
      "Loaded 26 RDF triples from https://nanocommons.github.io/datasets/\n",
      "Loaded 0 RDF triples from https://datamed.org/display-item.php?repository=0006&id=59139ef65152c62a9fc18ff7\n",
      "Loaded 7 RDF triples from https://bigcat-um.github.io/BiGCaT-Teaching/\n",
      "Invalid url: http://\n",
      "Loaded 78 RDF triples from https://www.ebi.ac.uk/interpro/entry/InterPro/IPR015252/\n",
      "Loaded 0 RDF triples from https://www.ensembl.org/id/ENSG00000139618\n",
      "Loaded 29 RDF triples from https://pcddb.cryst.bbk.ac.uk/deposit/CD0000031000\n",
      "Loaded 9 RDF triples from https://www.uniprot.org/uniprot/\n",
      "Loaded 0 RDF triples from https://ippidb.pasteur.fr/compounds/1103#compound\n",
      "Loaded 37 RDF triples from https://www.genenames.org/#data-catalog\n",
      "Loaded 77 RDF triples from https://www.ebi.ac.uk/training/events/metagenomics-bioinformatics-2022/\n",
      "Loaded 55 RDF triples from https://bioinformatics.ca/workshops-all/2021-infectious-disease-epidemiology-analysis/\n",
      "Loaded 14 RDF triples from https://hamap.expasy.org/archaea.html\n",
      "Loaded 12 RDF triples from https://www.ebi.ac.uk/Tools/hmmer/search/phmmer\n",
      "Loaded 8 RDF triples from https://biopragmatics.github.io/bioversions/\n",
      "Loaded 14 RDF triples from https://enzyme.expasy.org/EC/1.1.3.-\n",
      "Loaded 15 RDF triples from https://training.galaxyproject.org/training-material/hall-of-fame/hexylena/\n",
      "Loaded 14 RDF triples from http://gigadb.org/site/index\n",
      "Loaded 11 RDF triples from https://corkoakdb.org/gene/13014\n",
      "Loaded 213 RDF triples from https://swissmodel.expasy.org/repository\n",
      "Loaded 19 RDF triples from https://workflowhub.eu/people/9\n",
      "Loaded 132 RDF triples from https://www.ebi.ac.uk/chembl/\n",
      "Loaded 4 RDF triples from https://search.data.enanomapper.net/about/enanomapper/\n",
      "Loaded 54 RDF triples from https://reactome.org/content/detail/R-HSA-449147\n",
      "Loaded 12 RDF triples from https://www.guidetopharmacology.org/GRAC/LigandDisplayForward?ligandId=8299\n",
      "Loaded 5 RDF triples from https://bgee.org/?page=gene&gene_id=ENSG00000274928\n",
      "Loaded 7 RDF triples from https://scholia.toolforge.org/protein/Q424728\n",
      "Loaded 0 RDF triples from http://159.149.160.88/pscan/\n",
      "Loaded 143 RDF triples from https://www.sib.swiss/training/course/20221123_OPEMP\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded 20 RDF triples from https://www.elixir-europe.org/events/biohackathon-2018-paris\n",
      "Expecting value: line 1 column 1 (char 0)\n",
      "Loaded 0 RDF triples from https://pippa.psb.ugent.be/pippa_experiments/consult_experiment_basic_info/55\n",
      "Loaded 10 RDF triples from https://psnpbind.org/protein/1owh\n",
      "Loaded 44 RDF triples from http://www.cathdb.info\n",
      "Loaded 0 RDF triples from https://wikipathways.org/\n",
      "Loaded 35 RDF triples from https://bioschemas.org/meetings/2022-01_SWAT4HCLS_leiden\n",
      "Loaded 19 RDF triples from https://www.omicsdi.org/#/dataset/arrayexpress-repository/E-MTAB-6848\n",
      "Loaded 0 RDF triples from https://www.gbif.org/species/5220113\n",
      "Loaded 48 RDF triples from http://phenpath.biocomp.unibo.it/phenpath/\n",
      "Loaded 0 RDF triples from http://159.149.160.88/pscan_chip_dev/\n",
      "Loaded 27 RDF triples from https://identifiers.org/\n",
      "Loaded 31 RDF triples from https://www.ensembl.org/\n",
      "Loaded 49 RDF triples from https://zbmed-semtec.github.io/dome-galaxy-training/docs/DOME.html\n",
      "Loaded 46 RDF triples from https://zbmed-semtec.github.io/BioMedSem-IR-KD/docs/Lesson1.html\n",
      "Loaded 8 RDF triples from https://scholia.toolforge.org/gene/Q18030793\n",
      "Loaded 77 RDF triples from https://mobidb.org/P04637\n",
      "Loaded 0 RDF triples from https://www.france-bioinformatique.fr/en/evenements_upcoming\n",
      "Loaded 247 RDF triples from https://inpn.mnhn.fr/espece/cd_nom/60878/\n",
      "Loaded 44 RDF triples from https://www.guidetopharmacology.org/\n",
      "Loaded 182 RDF triples from https://string-db.org/\n",
      "Loaded 25 RDF triples from https://ega-archive.org/datasets/EGAD00000000001\n",
      "Loaded 35 RDF triples from https://tess.elixir-europe.org/events/1st-international-conference-on-fair-digital-objects-fdo2022\n",
      "Loaded 50 RDF triples from https://www.ebi.ac.uk/chembl/compound_report_card/CHEMBL59/\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>URL</th>\n",
       "      <th>HTTP_status</th>\n",
       "      <th>nb_triples</th>\n",
       "      <th>has_conforms_to</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>https://bridgedb.github.io/data/gene_database/</td>\n",
       "      <td>301.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>http://itsonedb.cloud.ba.infn.it/</td>\n",
       "      <td>200.0</td>\n",
       "      <td>95.0</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>https://www.sib.swiss/training/course/20230426...</td>\n",
       "      <td>200.0</td>\n",
       "      <td>124.0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>https://proteinensemble.org/PED00014</td>\n",
       "      <td>200.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>https://bioregistry.io/registry/hgnc</td>\n",
       "      <td>200.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>https://www.guidetopharmacology.org/</td>\n",
       "      <td>200.0</td>\n",
       "      <td>44.0</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132</th>\n",
       "      <td>https://string-db.org/</td>\n",
       "      <td>200.0</td>\n",
       "      <td>182.0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133</th>\n",
       "      <td>https://ega-archive.org/datasets/EGAD00000000001</td>\n",
       "      <td>200.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>134</th>\n",
       "      <td>https://tess.elixir-europe.org/events/1st-inte...</td>\n",
       "      <td>200.0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>https://www.ebi.ac.uk/chembl/compound_report_c...</td>\n",
       "      <td>200.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>136 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   URL  HTTP_status  \\\n",
       "0       https://bridgedb.github.io/data/gene_database/        301.0   \n",
       "1                    http://itsonedb.cloud.ba.infn.it/        200.0   \n",
       "2    https://www.sib.swiss/training/course/20230426...        200.0   \n",
       "3                 https://proteinensemble.org/PED00014        200.0   \n",
       "4                 https://bioregistry.io/registry/hgnc        200.0   \n",
       "..                                                 ...          ...   \n",
       "131               https://www.guidetopharmacology.org/        200.0   \n",
       "132                             https://string-db.org/        200.0   \n",
       "133   https://ega-archive.org/datasets/EGAD00000000001        200.0   \n",
       "134  https://tess.elixir-europe.org/events/1st-inte...        200.0   \n",
       "135  https://www.ebi.ac.uk/chembl/compound_report_c...        200.0   \n",
       "\n",
       "     nb_triples has_conforms_to  \n",
       "0          14.0            True  \n",
       "1          95.0           False  \n",
       "2         124.0            True  \n",
       "3           0.0           False  \n",
       "4          25.0            True  \n",
       "..          ...             ...  \n",
       "131        44.0           False  \n",
       "132       182.0            True  \n",
       "133        25.0           False  \n",
       "134        35.0            True  \n",
       "135        50.0           False  \n",
       "\n",
       "[136 rows x 4 columns]"
      ]
     },
     "execution_count": 188,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rows = []\n",
    "i = 0\n",
    "for ld in tqdm(lds):\n",
    "    row = {\"URL\": ld}    \n",
    "    try :\n",
    "        status_code = requests.head(ld).status_code\n",
    "        row[\"HTTP_status\"] = status_code\n",
    "        kg = retrieve_bioschemas(ld)\n",
    "        #print(kg.serialize(format=\"ntriples\"))\n",
    "        row[\"nb_triples\"] = len(kg)\n",
    "        row[\"has_conforms_to\"] = has_conforms_to(kg)\n",
    "    except InvalidURL as e :\n",
    "        print(f\"Invalid url: {ld}\")\n",
    "    rows.append(row)\n",
    "    \n",
    "df1 = pd.DataFrame.from_records(rows)\n",
    "df1.to_csv(\"bioschemas_harvesting.csv\")\n",
    "df1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f65bfb5e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "id": "d9027674",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['https://bridgedb.github.io/data/gene_database/', 'http://itsonedb.cloud.ba.infn.it/', 'https://www.sib.swiss/training/course/20230426_DOCK', 'https://proteinensemble.org/PED00014', 'https://bioregistry.io/registry/hgnc', 'https://humanmine.org/', 'https://modelarchive.org/', 'https://www.orpha.net/consor/cgi-bin/OC_Exp.php?Expert=141189&lng=en', 'https://deb-central.org/', 'https://bio.tools/blast', 'https://scholia.toolforge.org/chemical/Q18216', 'https://bmrb.io/data_library/summary/index.php?bmrbId=30309', 'https://bioschemas.org/tutorials/what_why_bioschemas', 'https://proteinensemble.org/', 'http://edgar.biocomp.unibo.it/', 'https://www.alliancegenome.org/gene/MGI:2442292', 'https://datacatalog.elixir-luxembourg.org/e/dataset/924dfe7a-71e8-11eb-bafe-3e22fbb3883f', 'https://www.metanetx.org/chem_info/MNXM680', 'https://nanocommons.github.io/tutorials/enteringData', 'https://zbmed.github.io/damalos/', 'https://swissmodel.expasy.org/', 'https://bridgedb.github.io/', 'https://www.birmingham.ac.uk/facilities/metabolomics-training-centre/courses/q-exactive.aspx', 'https://www.metanetx.org/', 'https://enanomapper.github.io/tutorials/Entering_and_analysing_nano_safety_data/readme.html', 'https://workflowhub.eu/', 'https://mint.bio.uniroma2.it/', 'https://bio.tools/covidmine', 'https://cancer.sanger.ac.uk/cosmic/', 'https://disprot.org/DP00086r026', 'https://www.covid19dataportal.org/', 'https://h2020-riskgone.github.io/riskgone-materials/identifiers_and_guidance.html', 'https://www.alliancegenome.org/gene/RGD:620474', 'https://workflowhub.eu/workflows/18', 'https://string-db.org/network/9606.ENSP00000379258', 'https://bgee.org/?page=species&amp;species_id=9606#expr-calls', 'https://prosite.expasy.org', 'http://www.ebi.ac.uk/pdbe/entry/pdb/4a7l', 'https://pathvisio.org/', 'https://fairsharing.org/', 'https://bar.biocomp.unibo.it/bar3/', 'https://pcddb.cryst.bbk.ac.uk/', 'https://www.ebi.ac.uk/biosamples/', 'https://massbank.eu/MassBank/RecordDisplay?id=LQB00001&dsn=RIKEN_IMS', 'https://pairedomicsdata.bioinformatics.nl/projects/d0f1add5-5b16-4f78-9b93-0d1d9c9bcb59.1', 'https://tess.elixir-europe.org/materials/hands-on-for-exome-sequencing-data-analysis-for-diagnosing-a-genetic-disease-tutorial', 'https://www.ebi.ac.uk/interpro/', 'https://cells.ebisc.org/BIHi006-D', 'https://rdmkit.elixir-europe.org/human_data.html', 'https://disprot.org/', 'https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/HGNC:37133', 'https://aip.fipapatients.org/', 'https://www.opalesurfcasting.net/connaissances/faune-et-flore/le-saumon-de-l-atlantique-salmo-salar.html', 'https://ippidb.pasteur.fr/', 'https://flymine.org/', 'https://tess.elixir-europe.org/events/introduction-to-r-for-biologists-online', 'https://zbmed.github.io/damalos/docs/2020.html', 'https://www.genenames.org/data/gene-symbol-report/#data-set', 'https://www.proteinatlas.org/', 'http://mvid-central.org/', 'https://disprot.org/DP00086', 'https://mobidb.org/', 'https://bgee.org/', 'https://www.chd7.org/', 'https://www.metanetx.org/ftp/', 'https://hpscreg.eu/cell-line/BIHi006-D', 'https://synbiohub.org/public/igem/BBa_K165005/1', 'https://ega-archive.org/', 'http://159.149.160.88/cscan', 'https://hamap.expasy.org/cgi-bin/unirule/unirule_browse.cgi', 'https://www.ebi.ac.uk/pdbe/entry/pdb/5hbi/protein/1', 'http://edgar.biocomp.unibo.it/cgi-bin/gene_disease_db/gene.py?gene=A2M', 'https://nanocommons.github.io/specifications/jrc/', 'https://www.gbif.org/dataset/c0cc29de-f49f-4b66-b4ec-c83afbb7101d', 'https://synbiohub.org/', 'https://pairedomicsdata.bioinformatics.nl/', 'https://www.guidetopharmacology.org/coronavirus.jsp', 'https://training.galaxyproject.org/training-material/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html', 'https://www.ebi.ac.uk/biosamples/samples/SAMEA491372', 'https://galaxyproject.org/events/', 'https://datacatalog.elixir-luxembourg.org/', 'https://scholia.toolforge.org/taxon/Q15978631', 'https://ega-archive.org/studies/EGAS00000000001', 'http://phenpath.biocomp.unibo.it/cgi-bin/phenpath/ppDB.py', 'https://nanocommons.github.io/datasets/', 'https://datamed.org/display-item.php?repository=0006&id=59139ef65152c62a9fc18ff7', 'https://bigcat-um.github.io/BiGCaT-Teaching/', 'http://', 'https://www.ebi.ac.uk/interpro/entry/InterPro/IPR015252/', 'https://www.ensembl.org/id/ENSG00000139618', 'https://pcddb.cryst.bbk.ac.uk/deposit/CD0000031000', 'https://www.uniprot.org/uniprot/', 'https://ippidb.pasteur.fr/compounds/1103#compound', 'https://www.genenames.org/#data-catalog', 'https://www.ebi.ac.uk/training/events/metagenomics-bioinformatics-2022/', 'https://bioinformatics.ca/workshops-all/2021-infectious-disease-epidemiology-analysis/', 'https://hamap.expasy.org/archaea.html', 'https://www.ebi.ac.uk/Tools/hmmer/search/phmmer', 'https://biopragmatics.github.io/bioversions/', 'https://enzyme.expasy.org/EC/1.1.3.-', 'https://training.galaxyproject.org/training-material/hall-of-fame/hexylena/', 'http://gigadb.org/site/index', 'https://corkoakdb.org/gene/13014', 'https://swissmodel.expasy.org/repository', 'https://workflowhub.eu/people/9', 'https://www.ebi.ac.uk/chembl/', 'https://search.data.enanomapper.net/about/enanomapper/', 'https://reactome.org/content/detail/R-HSA-449147', 'https://www.guidetopharmacology.org/GRAC/LigandDisplayForward?ligandId=8299', 'https://bgee.org/?page=gene&gene_id=ENSG00000274928', 'https://scholia.toolforge.org/protein/Q424728', 'http://159.149.160.88/pscan/', 'https://www.sib.swiss/training/course/20221123_OPEMP', 'https://www.elixir-europe.org/events/biohackathon-2018-paris', 'https://pippa.psb.ugent.be/pippa_experiments/consult_experiment_basic_info/55', 'https://psnpbind.org/protein/1owh', 'http://www.cathdb.info', 'https://wikipathways.org/', 'https://bioschemas.org/meetings/2022-01_SWAT4HCLS_leiden', 'https://www.omicsdi.org/#/dataset/arrayexpress-repository/E-MTAB-6848', 'https://www.gbif.org/species/5220113', 'http://phenpath.biocomp.unibo.it/phenpath/', 'http://159.149.160.88/pscan_chip_dev/', 'https://identifiers.org/', 'https://www.ensembl.org/', 'https://zbmed-semtec.github.io/dome-galaxy-training/docs/DOME.html', 'https://zbmed-semtec.github.io/BioMedSem-IR-KD/docs/Lesson1.html', 'https://scholia.toolforge.org/gene/Q18030793', 'https://mobidb.org/P04637', 'https://www.france-bioinformatique.fr/en/evenements_upcoming', 'https://inpn.mnhn.fr/espece/cd_nom/60878/', 'https://www.guidetopharmacology.org/', 'https://string-db.org/', 'https://ega-archive.org/datasets/EGAD00000000001', 'https://tess.elixir-europe.org/events/1st-international-conference-on-fair-digital-objects-fdo2022', 'https://www.ebi.ac.uk/chembl/compound_report_card/CHEMBL59/']\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "e7839853505447309b724839ba539d6d",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/136 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error while validating https://disprot.org/DP00086r026\n",
      "Error while validating https://disprot.org/DP00086\n",
      "Error while validating https://scholia.toolforge.org/taxon/Q15978631\n",
      "Error while validating http://\n",
      "Error while validating https://www.ensembl.org/id/ENSG00000139618\n",
      "Error while validating https://corkoakdb.org/gene/13014\n",
      "Error while validating https://pippa.psb.ugent.be/pippa_experiments/consult_experiment_basic_info/55\n",
      "Error while validating https://www.gbif.org/species/5220113\n",
      "Error while validating https://mobidb.org/P04637\n",
      "Error while validating https://inpn.mnhn.fr/espece/cd_nom/60878/\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Bioschemas Live Deploy URL</th>\n",
       "      <th>Evaluated entity</th>\n",
       "      <th>Reference profile</th>\n",
       "      <th>Is valid</th>\n",
       "      <th>Nb errors</th>\n",
       "      <th>Nb warnings</th>\n",
       "      <th>Is the latest profile</th>\n",
       "      <th>Is deprecated profile</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>https://bridgedb.github.io/data/gene_database/</td>\n",
       "      <td>https://bridgedb.github.io/data/gene_database/...</td>\n",
       "      <td>https://bioschemas.org/profiles/Dataset/1.0-RE...</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>https://www.sib.swiss/training/course/20230426...</td>\n",
       "      <td>https://www.sib.swiss/training/course/20231013...</td>\n",
       "      <td>https://bioschemas.org/profiles/CourseInstance...</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>https://www.sib.swiss/training/course/20230426...</td>\n",
       "      <td>https://www.sib.swiss/training/course/20230426...</td>\n",
       "      <td>https://bioschemas.org/profiles/CourseInstance...</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>https://www.sib.swiss/training/course/20230426...</td>\n",
       "      <td>https://sib-swiss.github.io/containers-introdu...</td>\n",
       "      <td>https://bioschemas.org/profiles/TrainingMateri...</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>https://www.sib.swiss/training/course/20230426...</td>\n",
       "      <td>Nc1d43fc4da0a48e7aabdec02e3633157</td>\n",
       "      <td>https://bioschemas.org/profiles/Course/1.0-REL...</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>https://www.ensembl.org/</td>\n",
       "      <td>http://www.ensembl.org/#project</td>\n",
       "      <td>https://bioschemas.org/profiles/DataCatalog/0....</td>\n",
       "      <td>False</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>https://zbmed-semtec.github.io/dome-galaxy-tra...</td>\n",
       "      <td>N425edc506725425e8e3e4459ad8aa342</td>\n",
       "      <td>https://bioschemas.org/profiles/TrainingMateri...</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>https://zbmed-semtec.github.io/BioMedSem-IR-KD...</td>\n",
       "      <td>N82cb704e1bf94a75af22d75f6de3e3b3</td>\n",
       "      <td>https://bioschemas.org/profiles/TrainingMateri...</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>https://scholia.toolforge.org/gene/Q18030793</td>\n",
       "      <td>N6cdc8012ac1b49ce9388e72b2db6ef88</td>\n",
       "      <td>https://bioschemas.org/profiles/Gene/0.7-RELEASE</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>https://bioschemas.org/profiles/Gene/1.0-RELEASE</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>https://tess.elixir-europe.org/events/1st-inte...</td>\n",
       "      <td>https://tess.elixir-europe.org/events/1st-inte...</td>\n",
       "      <td>https://bioschemas.org/profiles/Event/0.2-DRAF...</td>\n",
       "      <td>False</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>https://bioschemas.org/profiles/Event/0.3-DRAFT</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>78 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                           Bioschemas Live Deploy URL  \\\n",
       "0      https://bridgedb.github.io/data/gene_database/   \n",
       "1   https://www.sib.swiss/training/course/20230426...   \n",
       "2   https://www.sib.swiss/training/course/20230426...   \n",
       "3   https://www.sib.swiss/training/course/20230426...   \n",
       "4   https://www.sib.swiss/training/course/20230426...   \n",
       "..                                                ...   \n",
       "73                           https://www.ensembl.org/   \n",
       "74  https://zbmed-semtec.github.io/dome-galaxy-tra...   \n",
       "75  https://zbmed-semtec.github.io/BioMedSem-IR-KD...   \n",
       "76       https://scholia.toolforge.org/gene/Q18030793   \n",
       "77  https://tess.elixir-europe.org/events/1st-inte...   \n",
       "\n",
       "                                     Evaluated entity  \\\n",
       "0   https://bridgedb.github.io/data/gene_database/...   \n",
       "1   https://www.sib.swiss/training/course/20231013...   \n",
       "2   https://www.sib.swiss/training/course/20230426...   \n",
       "3   https://sib-swiss.github.io/containers-introdu...   \n",
       "4                   Nc1d43fc4da0a48e7aabdec02e3633157   \n",
       "..                                                ...   \n",
       "73                    http://www.ensembl.org/#project   \n",
       "74                  N425edc506725425e8e3e4459ad8aa342   \n",
       "75                  N82cb704e1bf94a75af22d75f6de3e3b3   \n",
       "76                  N6cdc8012ac1b49ce9388e72b2db6ef88   \n",
       "77  https://tess.elixir-europe.org/events/1st-inte...   \n",
       "\n",
       "                                    Reference profile  Is valid  Nb errors  \\\n",
       "0   https://bioschemas.org/profiles/Dataset/1.0-RE...      True          0   \n",
       "1   https://bioschemas.org/profiles/CourseInstance...      True          0   \n",
       "2   https://bioschemas.org/profiles/CourseInstance...      True          0   \n",
       "3   https://bioschemas.org/profiles/TrainingMateri...      True          0   \n",
       "4   https://bioschemas.org/profiles/Course/1.0-REL...      True          0   \n",
       "..                                                ...       ...        ...   \n",
       "73  https://bioschemas.org/profiles/DataCatalog/0....     False          1   \n",
       "74  https://bioschemas.org/profiles/TrainingMateri...      True          0   \n",
       "75  https://bioschemas.org/profiles/TrainingMateri...      True          0   \n",
       "76   https://bioschemas.org/profiles/Gene/0.7-RELEASE      True          0   \n",
       "77  https://bioschemas.org/profiles/Event/0.2-DRAF...     False          3   \n",
       "\n",
       "    Nb warnings                             Is the latest profile  \\\n",
       "0             9                                              True   \n",
       "1             0                                              True   \n",
       "2             0                                              True   \n",
       "3             8                                              True   \n",
       "4             8                                              True   \n",
       "..          ...                                               ...   \n",
       "73            6                                              True   \n",
       "74            6                                              True   \n",
       "75            6                                              True   \n",
       "76            3  https://bioschemas.org/profiles/Gene/1.0-RELEASE   \n",
       "77           10   https://bioschemas.org/profiles/Event/0.3-DRAFT   \n",
       "\n",
       "    Is deprecated profile  \n",
       "0                   False  \n",
       "1                   False  \n",
       "2                   False  \n",
       "3                   False  \n",
       "4                   False  \n",
       "..                    ...  \n",
       "73                  False  \n",
       "74                  False  \n",
       "75                  False  \n",
       "76                  False  \n",
       "77                  False  \n",
       "\n",
       "[78 rows x 8 columns]"
      ]
     },
     "execution_count": 187,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rows = []\n",
    "#print(lds)\n",
    "for ld in tqdm(lds):\n",
    "    try: \n",
    "        bs_valid = check_bioschemas(ld)\n",
    "        for entity in bs_valid.keys():\n",
    "            row = {\"Bioschemas Live Deploy URL\": ld}  \n",
    "            row[\"Evaluated entity\"] = entity\n",
    "            row[\"Reference profile\"] = bs_valid[entity][\"ref_profile\"]\n",
    "            row[\"Is valid\"] = bs_valid[entity][\"conforms\"]\n",
    "            row[\"Nb errors\"] = len(bs_valid[entity][\"errors\"])\n",
    "            row[\"Nb warnings\"] = len(bs_valid[entity][\"warnings\"])\n",
    "            row[\"Is the latest profile\"] = bs_valid[entity][\"latest_profile\"]\n",
    "            row[\"Is deprecated profile\"] = bs_valid[entity][\"deprecated\"]\n",
    "            rows.append(row)\n",
    "    except Exception as e: \n",
    "        print(f\"Error while validating {ld}\")\n",
    "df2 = pd.DataFrame.from_records(rows)\n",
    "df2.to_csv(\"bioschemas_validation.csv\")\n",
    "df2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c59c5820",
   "metadata": {},
   "outputs": [],
   "source": [
    "from IPython.display import display, Markdown, Latex\n",
    "my_md = '<i class=\"fa fa-camera\"></i>'\n",
    "display(Markdown(my_md))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "503d0126",
   "metadata": {},
   "outputs": [],
   "source": [
    "md1 = \"\"\"\n",
    "{\"@id\": \"_:N0b2cabb67c8a461a8c1c84367ef89fb1\", \"http://www.w3.org/1999/xhtml/vocab#role\": [{\"@id\": \"http://www.w3.org/1999/xhtml/vocab#img\"}]}\n",
    "\"\"\"\n",
    "\n",
    "md2 = \"\"\"\n",
    "{\"type\": \"http://schema.org/Organization\", \"properties\": {\"logo\": \"http://bioschemas.org/assets/img/square_logo2.png\", \"url\": \"http://bioschemas.org\", \"email\": \"enquiries@bioschemas.org\", \"name\": \"Bioschemas Community\"}}\n",
    "\"\"\"\n",
    "\n",
    "md3 = \"\"\"\n",
    "\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "id": "f19925e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "#live_deploys_remote_file = \"https://raw.githubusercontent.com/BioSchemas/bioschemas.github.io/master/_data/live_deployments.json\"\n",
    "#res = requests.get(live_deploys_remote_file)\n",
    "#live_deploys = res.json()\n",
    "#print(json.dumps(live_deploys, indent=True))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "id": "e2b87c87",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{\n",
      " \"https://biocomputingup.it/#Organization\": {\n",
      "  \"method\": \"by_conformsto\",\n",
      "  \"type\": \"http://schema.org/Organization\",\n",
      "  \"ref_profile\": \"https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19\",\n",
      "  \"conforms\": false,\n",
      "  \"warnings\": [\n",
      "   \"https://schema.org/alternateName\",\n",
      "   \"https://schema.org/contactPoint\",\n",
      "   \"https://discovery.biothings.io/view/bioschemas/fundingModel\",\n",
      "   \"https://schema.org/keywords\",\n",
      "   \"https://schema.org/location\",\n",
      "   \"https://schema.org/logo\",\n",
      "   \"https://schema.org/member\",\n",
      "   \"https://schema.org/memberOf\",\n",
      "   \"https://discovery.biothings.io/view/bioschemas/membershipCategory\",\n",
      "   \"https://discovery.biothings.io/view/bioschemas/status\",\n",
      "   \"https://schema.org/url\"\n",
      "  ],\n",
      "  \"errors\": [\n",
      "   \"https://discovery.biothings.io/view/bioschemas/topic\"\n",
      "  ],\n",
      "  \"deprecated\": false,\n",
      "  \"latest_profile\": \"https://bioschemas.org/profiles/Organization/0.3-DRAFT\"\n",
      " },\n",
      " \"https://mobidb.org/#DataCatalog\": {\n",
      "  \"method\": \"by_conformsto\",\n",
      "  \"type\": \"http://schema.org/DataCatalog\",\n",
      "  \"ref_profile\": \"https://bioschemas.org/profiles/DataCatalog/0.3-RELEASE-2019_07_01\",\n",
      "  \"conforms\": true,\n",
      "  \"warnings\": [\n",
      "   \"https://schema.org/about\",\n",
      "   \"https://schema.org/alternateName\",\n",
      "   \"https://schema.org/dateCreated\"\n",
      "  ],\n",
      "  \"errors\": [],\n",
      "  \"deprecated\": false,\n",
      "  \"latest_profile\": true\n",
      " }\n",
      "}\n"
     ]
    }
   ],
   "source": [
    "mobidb_valid = check_bioschemas(\"https://mobidb.org\")\n",
    "print(json.dumps(mobidb_valid, indent=True))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 179,
   "id": "94e29a4a",
   "metadata": {},
   "outputs": [
    {
     "ename": "BadSyntax",
     "evalue": "at line 4 of <>:\nBad syntax (expected '.' or '}' or ']' at end of statement) at ^ in:\n\"b'<!DOCTYPE html>\\n<html lang=\"en\">\\n<html>\\n  '^b'<head>\\n  <meta charset=\"utf-8\">\\n  <meta http-equiv=\"X-UA-Com'...\"",
     "output_type": "error",
     "traceback": [
      "Traceback \u001b[0;36m(most recent call last)\u001b[0m:\n",
      "  File \u001b[1;32m~/miniconda3/envs/fair-checker/lib/python3.9/site-packages/IPython/core/interactiveshell.py:3361\u001b[0m in \u001b[1;35mrun_code\u001b[0m\n    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  Input \u001b[1;32mIn [179]\u001b[0m in \u001b[1;35m<cell line: 2>\u001b[0m\n    g.parse(\"http://bioschemas.org/\", format=\"n3\")\n",
      "  File \u001b[1;32m~/miniconda3/envs/fair-checker/lib/python3.9/site-packages/rdflib/graph.py:1977\u001b[0m in \u001b[1;35mparse\u001b[0m\n    context.parse(source, publicID=publicID, format=format, **args)\n",
      "  File \u001b[1;32m~/miniconda3/envs/fair-checker/lib/python3.9/site-packages/rdflib/graph.py:1339\u001b[0m in \u001b[1;35mparse\u001b[0m\n    raise se\n",
      "  File \u001b[1;32m~/miniconda3/envs/fair-checker/lib/python3.9/site-packages/rdflib/graph.py:1330\u001b[0m in \u001b[1;35mparse\u001b[0m\n    parser.parse(source, self, **args)\n",
      "  File \u001b[1;32m~/miniconda3/envs/fair-checker/lib/python3.9/site-packages/rdflib/plugins/parsers/notation3.py:1977\u001b[0m in \u001b[1;35mparse\u001b[0m\n    TurtleParser.parse(self, source, conj_graph, encoding, turtle=False)\n",
      "  File \u001b[1;32m~/miniconda3/envs/fair-checker/lib/python3.9/site-packages/rdflib/plugins/parsers/notation3.py:1943\u001b[0m in \u001b[1;35mparse\u001b[0m\n    p.loadStream(stream)\n",
      "  File \u001b[1;32m~/miniconda3/envs/fair-checker/lib/python3.9/site-packages/rdflib/plugins/parsers/notation3.py:454\u001b[0m in \u001b[1;35mloadStream\u001b[0m\n    return self.loadBuf(stream.read())  # Not ideal\n",
      "  File \u001b[1;32m~/miniconda3/envs/fair-checker/lib/python3.9/site-packages/rdflib/plugins/parsers/notation3.py:460\u001b[0m in \u001b[1;35mloadBuf\u001b[0m\n    self.feed(buf)\n",
      "  File \u001b[1;32m~/miniconda3/envs/fair-checker/lib/python3.9/site-packages/rdflib/plugins/parsers/notation3.py:486\u001b[0m in \u001b[1;35mfeed\u001b[0m\n    i = self.directiveOrStatement(s, j)\n",
      "  File \u001b[1;32m~/miniconda3/envs/fair-checker/lib/python3.9/site-packages/rdflib/plugins/parsers/notation3.py:508\u001b[0m in \u001b[1;35mdirectiveOrStatement\u001b[0m\n    return self.checkDot(argstr, j)\n",
      "  File \u001b[1;32m~/miniconda3/envs/fair-checker/lib/python3.9/site-packages/rdflib/plugins/parsers/notation3.py:1169\u001b[0m in \u001b[1;35mcheckDot\u001b[0m\n    self.BadSyntax(argstr, j, \"expected '.' or '}' or ']' at end of statement\")\n",
      "\u001b[0;36m  File \u001b[0;32m~/miniconda3/envs/fair-checker/lib/python3.9/site-packages/rdflib/plugins/parsers/notation3.py:1666\u001b[0;36m in \u001b[0;35mBadSyntax\u001b[0;36m\u001b[0m\n\u001b[0;31m    raise BadSyntax(self._thisDoc, self.lines, argstr, i, msg)\u001b[0m\n",
      "\u001b[0;36m  File \u001b[0;32m<string>\u001b[0;36m\u001b[0m\n\u001b[0;31mBadSyntax\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "g = ConjunctiveGraph()\n",
    "g.parse(\"http://bioschemas.org/\", format=\"n3\")\n",
    "#print(g.serialize(format=\"turtle\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f7b46364",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
